// meraki_dashboard_api/src/model/Body24.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body24 model module.
* @module model/Body24
* @version 1.0.0
*/
export default class Body24 {
    /**
    * Constructs a new <code>Body24</code>.
    * @alias module:model/Body24
    * @class
    * @param access {module:model/Body24.AccessEnum} A string indicating the rule for which IPs are allowed to use the specified service. Acceptable values are \"blocked\" (no remote IPs can access the service), \"restricted\" (only allowed IPs can access the service), and \"unrestriced\" (any remote IP can access the service). This field is required
    */

    constructor(access) {
        
        
        this['access'] = access;
        
    }

    /**
    * Constructs a <code>Body24</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body24} obj Optional instance to populate.
    * @return {module:model/Body24} The populated <code>Body24</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body24();
                        
            
            if (data.hasOwnProperty('access')) {
                obj['access'] = ApiClient.convertToType(data['access'], 'String');
            }
            if (data.hasOwnProperty('allowedIps')) {
                obj['allowedIps'] = ApiClient.convertToType(data['allowedIps'], ['String']);
            }
        }
        return obj;
    }

    /**
    * A string indicating the rule for which IPs are allowed to use the specified service. Acceptable values are \"blocked\" (no remote IPs can access the service), \"restricted\" (only allowed IPs can access the service), and \"unrestriced\" (any remote IP can access the service). This field is required
    * @member {module:model/Body24.AccessEnum} access
    */
    'access' = undefined;
    /**
    * An array of allowed IPs that can access the service. This field is required if \"access\" is set to \"restricted\". Otherwise this field is ignored
    * @member {Array.<String>} allowedIps
    */
    'allowedIps' = undefined;



    /**
    * Allowed values for the <code>access</code> property.
    * @enum {String}
    * @readonly
    */
    static AccessEnum = {
        /**
         * value: "blocked"
         * @const
         */
        "blocked": "blocked",
        /**
         * value: "restricted"
         * @const
         */
        "restricted": "restricted",
        /**
         * value: "unrestricted"
         * @const
         */
        "unrestricted": "unrestricted"    };

}
