// meraki_dashboard_api/src/model/Body139.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body139 model module.
* @module model/Body139
* @version 1.0.0
*/
export default class Body139 {
    /**
    * Constructs a new <code>Body139</code>.
    * @alias module:model/Body139
    * @class
    * @param licenseId {String} The ID of the SM license to assign seats from
    * @param networkId {String} The ID of the SM network to assign the seats to
    * @param seatCount {Number} The number of seats to assign to the SM network. Must be less than or equal to the total number of seats of the license
    */

    constructor(licenseId, networkId, seatCount) {
        
        
        this['licenseId'] = licenseId;
        this['networkId'] = networkId;
        this['seatCount'] = seatCount;
        
    }

    /**
    * Constructs a <code>Body139</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body139} obj Optional instance to populate.
    * @return {module:model/Body139} The populated <code>Body139</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body139();
                        
            
            if (data.hasOwnProperty('licenseId')) {
                obj['licenseId'] = ApiClient.convertToType(data['licenseId'], 'String');
            }
            if (data.hasOwnProperty('networkId')) {
                obj['networkId'] = ApiClient.convertToType(data['networkId'], 'String');
            }
            if (data.hasOwnProperty('seatCount')) {
                obj['seatCount'] = ApiClient.convertToType(data['seatCount'], 'Number');
            }
        }
        return obj;
    }

    /**
    * The ID of the SM license to assign seats from
    * @member {String} licenseId
    */
    'licenseId' = undefined;
    /**
    * The ID of the SM network to assign the seats to
    * @member {String} networkId
    */
    'networkId' = undefined;
    /**
    * The number of seats to assign to the SM network. Must be less than or equal to the total number of seats of the license
    * @member {Number} seatCount
    */
    'seatCount' = undefined;




}
