// meraki_dashboard_api/src/model/Body20.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdalertssettingsAlerts from './NetworksnetworkIdalertssettingsAlerts';
import NetworksnetworkIdalertssettingsDefaultDestinations from './NetworksnetworkIdalertssettingsDefaultDestinations';

/**
* The Body20 model module.
* @module model/Body20
* @version 1.0.0
*/
export default class Body20 {
    /**
    * Constructs a new <code>Body20</code>.
    * @alias module:model/Body20
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body20</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body20} obj Optional instance to populate.
    * @return {module:model/Body20} The populated <code>Body20</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body20();
                        
            
            if (data.hasOwnProperty('defaultDestinations')) {
                obj['defaultDestinations'] = NetworksnetworkIdalertssettingsDefaultDestinations.constructFromObject(data['defaultDestinations']);
            }
            if (data.hasOwnProperty('alerts')) {
                obj['alerts'] = ApiClient.convertToType(data['alerts'], [NetworksnetworkIdalertssettingsAlerts]);
            }
        }
        return obj;
    }

    /**
    * @member {module:model/NetworksnetworkIdalertssettingsDefaultDestinations} defaultDestinations
    */
    'defaultDestinations' = undefined;
    /**
    * Alert-specific configuration for each type. Only alerts that pertain to the network can be updated.
    * @member {Array.<module:model/NetworksnetworkIdalertssettingsAlerts>} alerts
    */
    'alerts' = undefined;




}
