// meraki_dashboard_api/src/model/Body82.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdswitchmtuOverrides from './NetworksnetworkIdswitchmtuOverrides';

/**
* The Body82 model module.
* @module model/Body82
* @version 1.0.0
*/
export default class Body82 {
    /**
    * Constructs a new <code>Body82</code>.
    * @alias module:model/Body82
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body82</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body82} obj Optional instance to populate.
    * @return {module:model/Body82} The populated <code>Body82</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body82();
                        
            
            if (data.hasOwnProperty('defaultMtuSize')) {
                obj['defaultMtuSize'] = ApiClient.convertToType(data['defaultMtuSize'], 'Number');
            }
            if (data.hasOwnProperty('overrides')) {
                obj['overrides'] = ApiClient.convertToType(data['overrides'], [NetworksnetworkIdswitchmtuOverrides]);
            }
        }
        return obj;
    }

    /**
    * MTU size for the entire network. Default value is 9578.
    * @member {Number} defaultMtuSize
    */
    'defaultMtuSize' = undefined;
    /**
    * Override MTU size for individual switches or switch profiles. An empty array will clear overrides.
    * @member {Array.<module:model/NetworksnetworkIdswitchmtuOverrides>} overrides
    */
    'overrides' = undefined;




}
