// meraki_dashboard_api/src/model/Body80.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdswitchlinkAggregationsSwitchPorts from './NetworksnetworkIdswitchlinkAggregationsSwitchPorts';
import NetworksnetworkIdswitchlinkAggregationsSwitchProfilePorts from './NetworksnetworkIdswitchlinkAggregationsSwitchProfilePorts';

/**
* The Body80 model module.
* @module model/Body80
* @version 1.0.0
*/
export default class Body80 {
    /**
    * Constructs a new <code>Body80</code>.
    * @alias module:model/Body80
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body80</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body80} obj Optional instance to populate.
    * @return {module:model/Body80} The populated <code>Body80</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body80();
                        
            
            if (data.hasOwnProperty('switchPorts')) {
                obj['switchPorts'] = ApiClient.convertToType(data['switchPorts'], [NetworksnetworkIdswitchlinkAggregationsSwitchPorts]);
            }
            if (data.hasOwnProperty('switchProfilePorts')) {
                obj['switchProfilePorts'] = ApiClient.convertToType(data['switchProfilePorts'], [NetworksnetworkIdswitchlinkAggregationsSwitchProfilePorts]);
            }
        }
        return obj;
    }

    /**
    * Array of switch or stack ports for creating aggregation group. Minimum 2 and maximum 8 ports are supported.
    * @member {Array.<module:model/NetworksnetworkIdswitchlinkAggregationsSwitchPorts>} switchPorts
    */
    'switchPorts' = undefined;
    /**
    * Array of switch profile ports for creating aggregation group. Minimum 2 and maximum 8 ports are supported.
    * @member {Array.<module:model/NetworksnetworkIdswitchlinkAggregationsSwitchProfilePorts>} switchProfilePorts
    */
    'switchProfilePorts' = undefined;




}
