// meraki_dashboard_api/src/model/Body123.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body123 model module.
* @module model/Body123
* @version 1.0.0
*/
export default class Body123 {
    /**
    * Constructs a new <code>Body123</code>.
    * @alias module:model/Body123
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body123</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body123} obj Optional instance to populate.
    * @return {module:model/Body123} The populated <code>Body123</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body123();
                        
            
            if (data.hasOwnProperty('confirmed')) {
                obj['confirmed'] = ApiClient.convertToType(data['confirmed'], 'Boolean');
            }
            if (data.hasOwnProperty('synchronous')) {
                obj['synchronous'] = ApiClient.convertToType(data['synchronous'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * A boolean representing whether or not the batch has been confirmed. This property cannot be unset once it is true.
    * @member {Boolean} confirmed
    */
    'confirmed' = undefined;
    /**
    * Set to true to force the batch to run synchronous. There can be at most 20 actions in synchronous batch.
    * @member {Boolean} synchronous
    */
    'synchronous' = undefined;




}
