// meraki_dashboard_api/src/model/Body41.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences from './NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences';
import NetworksnetworkIdappliancetrafficShapinguplinkSelectionWanTrafficUplinkPreferences from './NetworksnetworkIdappliancetrafficShapinguplinkSelectionWanTrafficUplinkPreferences';

/**
* The Body41 model module.
* @module model/Body41
* @version 1.0.0
*/
export default class Body41 {
    /**
    * Constructs a new <code>Body41</code>.
    * @alias module:model/Body41
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body41</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body41} obj Optional instance to populate.
    * @return {module:model/Body41} The populated <code>Body41</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body41();
                        
            
            if (data.hasOwnProperty('activeActiveAutoVpnEnabled')) {
                obj['activeActiveAutoVpnEnabled'] = ApiClient.convertToType(data['activeActiveAutoVpnEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('defaultUplink')) {
                obj['defaultUplink'] = ApiClient.convertToType(data['defaultUplink'], 'String');
            }
            if (data.hasOwnProperty('loadBalancingEnabled')) {
                obj['loadBalancingEnabled'] = ApiClient.convertToType(data['loadBalancingEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('wanTrafficUplinkPreferences')) {
                obj['wanTrafficUplinkPreferences'] = ApiClient.convertToType(data['wanTrafficUplinkPreferences'], [NetworksnetworkIdappliancetrafficShapinguplinkSelectionWanTrafficUplinkPreferences]);
            }
            if (data.hasOwnProperty('vpnTrafficUplinkPreferences')) {
                obj['vpnTrafficUplinkPreferences'] = ApiClient.convertToType(data['vpnTrafficUplinkPreferences'], [NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences]);
            }
        }
        return obj;
    }

    /**
    * Toggle for enabling or disabling active-active AutoVPN
    * @member {Boolean} activeActiveAutoVpnEnabled
    */
    'activeActiveAutoVpnEnabled' = undefined;
    /**
    * The default uplink. Must be one of: 'wan1' or 'wan2'
    * @member {module:model/Body41.DefaultUplinkEnum} defaultUplink
    */
    'defaultUplink' = undefined;
    /**
    * Toggle for enabling or disabling load balancing
    * @member {Boolean} loadBalancingEnabled
    */
    'loadBalancingEnabled' = undefined;
    /**
    * Array of uplink preference rules for WAN traffic
    * @member {Array.<module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionWanTrafficUplinkPreferences>} wanTrafficUplinkPreferences
    */
    'wanTrafficUplinkPreferences' = undefined;
    /**
    * Array of uplink preference rules for VPN traffic
    * @member {Array.<module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences>} vpnTrafficUplinkPreferences
    */
    'vpnTrafficUplinkPreferences' = undefined;



    /**
    * Allowed values for the <code>defaultUplink</code> property.
    * @enum {String}
    * @readonly
    */
    static DefaultUplinkEnum = {
        /**
         * value: "wan1"
         * @const
         */
        "wan1": "wan1",
        /**
         * value: "wan2"
         * @const
         */
        "wan2": "wan2"    };

}
