// meraki_dashboard_api/src/model/Body22.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body22 model module.
* @module model/Body22
* @version 1.0.0
*/
export default class Body22 {
    /**
    * Constructs a new <code>Body22</code>.
    * @alias module:model/Body22
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body22</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body22} obj Optional instance to populate.
    * @return {module:model/Body22} The populated <code>Body22</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body22();
                        
            
            if (data.hasOwnProperty('allowedUrlPatterns')) {
                obj['allowedUrlPatterns'] = ApiClient.convertToType(data['allowedUrlPatterns'], ['String']);
            }
            if (data.hasOwnProperty('blockedUrlPatterns')) {
                obj['blockedUrlPatterns'] = ApiClient.convertToType(data['blockedUrlPatterns'], ['String']);
            }
            if (data.hasOwnProperty('blockedUrlCategories')) {
                obj['blockedUrlCategories'] = ApiClient.convertToType(data['blockedUrlCategories'], ['String']);
            }
            if (data.hasOwnProperty('urlCategoryListSize')) {
                obj['urlCategoryListSize'] = ApiClient.convertToType(data['urlCategoryListSize'], 'String');
            }
        }
        return obj;
    }

    /**
    * A list of URL patterns that are allowed
    * @member {Array.<String>} allowedUrlPatterns
    */
    'allowedUrlPatterns' = undefined;
    /**
    * A list of URL patterns that are blocked
    * @member {Array.<String>} blockedUrlPatterns
    */
    'blockedUrlPatterns' = undefined;
    /**
    * A list of URL categories to block
    * @member {Array.<String>} blockedUrlCategories
    */
    'blockedUrlCategories' = undefined;
    /**
    * URL category list size which is either 'topSites' or 'fullList'
    * @member {module:model/Body22.UrlCategoryListSizeEnum} urlCategoryListSize
    */
    'urlCategoryListSize' = undefined;



    /**
    * Allowed values for the <code>urlCategoryListSize</code> property.
    * @enum {String}
    * @readonly
    */
    static UrlCategoryListSizeEnum = {
        /**
         * value: "topSites"
         * @const
         */
        "topSites": "topSites",
        /**
         * value: "fullList"
         * @const
         */
        "fullList": "fullList"    };

}
