// meraki_dashboard_api/src/model/Body113.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdwirelessssidsnumberApTagsAndVlanIds from './NetworksnetworkIdwirelessssidsnumberApTagsAndVlanIds';
import NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers from './NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers';
import NetworksnetworkIdwirelessssidsnumberRadiusServers from './NetworksnetworkIdwirelessssidsnumberRadiusServers';

/**
* The Body113 model module.
* @module model/Body113
* @version 1.0.0
*/
export default class Body113 {
    /**
    * Constructs a new <code>Body113</code>.
    * @alias module:model/Body113
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body113</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body113} obj Optional instance to populate.
    * @return {module:model/Body113} The populated <code>Body113</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body113();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('enabled')) {
                obj['enabled'] = ApiClient.convertToType(data['enabled'], 'Boolean');
            }
            if (data.hasOwnProperty('authMode')) {
                obj['authMode'] = ApiClient.convertToType(data['authMode'], 'String');
            }
            if (data.hasOwnProperty('enterpriseAdminAccess')) {
                obj['enterpriseAdminAccess'] = ApiClient.convertToType(data['enterpriseAdminAccess'], 'String');
            }
            if (data.hasOwnProperty('encryptionMode')) {
                obj['encryptionMode'] = ApiClient.convertToType(data['encryptionMode'], 'String');
            }
            if (data.hasOwnProperty('psk')) {
                obj['psk'] = ApiClient.convertToType(data['psk'], 'String');
            }
            if (data.hasOwnProperty('wpaEncryptionMode')) {
                obj['wpaEncryptionMode'] = ApiClient.convertToType(data['wpaEncryptionMode'], 'String');
            }
            if (data.hasOwnProperty('splashPage')) {
                obj['splashPage'] = ApiClient.convertToType(data['splashPage'], 'String');
            }
            if (data.hasOwnProperty('radiusServers')) {
                obj['radiusServers'] = ApiClient.convertToType(data['radiusServers'], [NetworksnetworkIdwirelessssidsnumberRadiusServers]);
            }
            if (data.hasOwnProperty('radiusProxyEnabled')) {
                obj['radiusProxyEnabled'] = ApiClient.convertToType(data['radiusProxyEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('radiusCoaEnabled')) {
                obj['radiusCoaEnabled'] = ApiClient.convertToType(data['radiusCoaEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('radiusFailoverPolicy')) {
                obj['radiusFailoverPolicy'] = ApiClient.convertToType(data['radiusFailoverPolicy'], 'String');
            }
            if (data.hasOwnProperty('radiusLoadBalancingPolicy')) {
                obj['radiusLoadBalancingPolicy'] = ApiClient.convertToType(data['radiusLoadBalancingPolicy'], 'String');
            }
            if (data.hasOwnProperty('radiusAccountingEnabled')) {
                obj['radiusAccountingEnabled'] = ApiClient.convertToType(data['radiusAccountingEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('radiusAccountingServers')) {
                obj['radiusAccountingServers'] = ApiClient.convertToType(data['radiusAccountingServers'], [NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers]);
            }
            if (data.hasOwnProperty('radiusAttributeForGroupPolicies')) {
                obj['radiusAttributeForGroupPolicies'] = ApiClient.convertToType(data['radiusAttributeForGroupPolicies'], 'String');
            }
            if (data.hasOwnProperty('ipAssignmentMode')) {
                obj['ipAssignmentMode'] = ApiClient.convertToType(data['ipAssignmentMode'], 'String');
            }
            if (data.hasOwnProperty('useVlanTagging')) {
                obj['useVlanTagging'] = ApiClient.convertToType(data['useVlanTagging'], 'Boolean');
            }
            if (data.hasOwnProperty('concentratorNetworkId')) {
                obj['concentratorNetworkId'] = ApiClient.convertToType(data['concentratorNetworkId'], 'String');
            }
            if (data.hasOwnProperty('vlanId')) {
                obj['vlanId'] = ApiClient.convertToType(data['vlanId'], 'Number');
            }
            if (data.hasOwnProperty('defaultVlanId')) {
                obj['defaultVlanId'] = ApiClient.convertToType(data['defaultVlanId'], 'Number');
            }
            if (data.hasOwnProperty('apTagsAndVlanIds')) {
                obj['apTagsAndVlanIds'] = ApiClient.convertToType(data['apTagsAndVlanIds'], [NetworksnetworkIdwirelessssidsnumberApTagsAndVlanIds]);
            }
            if (data.hasOwnProperty('walledGardenEnabled')) {
                obj['walledGardenEnabled'] = ApiClient.convertToType(data['walledGardenEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('walledGardenRanges')) {
                obj['walledGardenRanges'] = ApiClient.convertToType(data['walledGardenRanges'], ['String']);
            }
            if (data.hasOwnProperty('radiusOverride')) {
                obj['radiusOverride'] = ApiClient.convertToType(data['radiusOverride'], 'Boolean');
            }
            if (data.hasOwnProperty('radiusGuestVlanEnabled')) {
                obj['radiusGuestVlanEnabled'] = ApiClient.convertToType(data['radiusGuestVlanEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('radiusGuestVlanId')) {
                obj['radiusGuestVlanId'] = ApiClient.convertToType(data['radiusGuestVlanId'], 'Number');
            }
            if (data.hasOwnProperty('minBitrate')) {
                obj['minBitrate'] = ApiClient.convertToType(data['minBitrate'], 'Number');
            }
            if (data.hasOwnProperty('bandSelection')) {
                obj['bandSelection'] = ApiClient.convertToType(data['bandSelection'], 'String');
            }
            if (data.hasOwnProperty('perClientBandwidthLimitUp')) {
                obj['perClientBandwidthLimitUp'] = ApiClient.convertToType(data['perClientBandwidthLimitUp'], 'Number');
            }
            if (data.hasOwnProperty('perClientBandwidthLimitDown')) {
                obj['perClientBandwidthLimitDown'] = ApiClient.convertToType(data['perClientBandwidthLimitDown'], 'Number');
            }
            if (data.hasOwnProperty('perSsidBandwidthLimitUp')) {
                obj['perSsidBandwidthLimitUp'] = ApiClient.convertToType(data['perSsidBandwidthLimitUp'], 'Number');
            }
            if (data.hasOwnProperty('perSsidBandwidthLimitDown')) {
                obj['perSsidBandwidthLimitDown'] = ApiClient.convertToType(data['perSsidBandwidthLimitDown'], 'Number');
            }
            if (data.hasOwnProperty('lanIsolationEnabled')) {
                obj['lanIsolationEnabled'] = ApiClient.convertToType(data['lanIsolationEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('visible')) {
                obj['visible'] = ApiClient.convertToType(data['visible'], 'Boolean');
            }
            if (data.hasOwnProperty('availableOnAllAps')) {
                obj['availableOnAllAps'] = ApiClient.convertToType(data['availableOnAllAps'], 'Boolean');
            }
            if (data.hasOwnProperty('availabilityTags')) {
                obj['availabilityTags'] = ApiClient.convertToType(data['availabilityTags'], ['String']);
            }
        }
        return obj;
    }

    /**
    * The name of the SSID
    * @member {String} name
    */
    'name' = undefined;
    /**
    * Whether or not the SSID is enabled
    * @member {Boolean} enabled
    */
    'enabled' = undefined;
    /**
    * The association control method for the SSID ('open', 'psk', 'open-with-radius', '8021x-meraki', '8021x-radius', 'ipsk-with-radius' or 'ipsk-without-radius')
    * @member {module:model/Body113.AuthModeEnum} authMode
    */
    'authMode' = undefined;
    /**
    * Whether or not an SSID is accessible by 'enterprise' administrators ('access disabled' or 'access enabled')
    * @member {module:model/Body113.EnterpriseAdminAccessEnum} enterpriseAdminAccess
    */
    'enterpriseAdminAccess' = undefined;
    /**
    * The psk encryption mode for the SSID ('wep' or 'wpa'). This param is only valid if the authMode is 'psk'
    * @member {module:model/Body113.EncryptionModeEnum} encryptionMode
    */
    'encryptionMode' = undefined;
    /**
    * The passkey for the SSID. This param is only valid if the authMode is 'psk'
    * @member {String} psk
    */
    'psk' = undefined;
    /**
    * The types of WPA encryption. ('WPA1 only', 'WPA1 and WPA2', 'WPA2 only', 'WPA3 Transition Mode' or 'WPA3 only')
    * @member {module:model/Body113.WpaEncryptionModeEnum} wpaEncryptionMode
    */
    'wpaEncryptionMode' = undefined;
    /**
    * The type of splash page for the SSID ('None', 'Click-through splash page', 'Billing', 'Password-protected with Meraki RADIUS', 'Password-protected with custom RADIUS', 'Password-protected with Active Directory', 'Password-protected with LDAP', 'SMS authentication', 'Systems Manager Sentry', 'Facebook Wi-Fi', 'Google OAuth' or 'Sponsored guest'). This attribute is not supported for template children.
    * @member {module:model/Body113.SplashPageEnum} splashPage
    */
    'splashPage' = undefined;
    /**
    * The RADIUS 802.1X servers to be used for authentication. This param is only valid if the authMode is 'open-with-radius', '8021x-radius' or 'ipsk-with-radius'
    * @member {Array.<module:model/NetworksnetworkIdwirelessssidsnumberRadiusServers>} radiusServers
    */
    'radiusServers' = undefined;
    /**
    * If true, Meraki devices will proxy RADIUS messages through the Meraki cloud to the configured RADIUS auth and accounting servers.
    * @member {Boolean} radiusProxyEnabled
    */
    'radiusProxyEnabled' = undefined;
    /**
    * If true, Meraki devices will act as a RADIUS Dynamic Authorization Server and will respond to RADIUS Change-of-Authorization and Disconnect messages sent by the RADIUS server.
    * @member {Boolean} radiusCoaEnabled
    */
    'radiusCoaEnabled' = undefined;
    /**
    * This policy determines how authentication requests should be handled in the event that all of the configured RADIUS servers are unreachable ('Deny access' or 'Allow access')
    * @member {module:model/Body113.RadiusFailoverPolicyEnum} radiusFailoverPolicy
    */
    'radiusFailoverPolicy' = undefined;
    /**
    * This policy determines which RADIUS server will be contacted first in an authentication attempt and the ordering of any necessary retry attempts ('Strict priority order' or 'Round robin')
    * @member {module:model/Body113.RadiusLoadBalancingPolicyEnum} radiusLoadBalancingPolicy
    */
    'radiusLoadBalancingPolicy' = undefined;
    /**
    * Whether or not RADIUS accounting is enabled. This param is only valid if the authMode is 'open-with-radius', '8021x-radius' or 'ipsk-with-radius'
    * @member {Boolean} radiusAccountingEnabled
    */
    'radiusAccountingEnabled' = undefined;
    /**
    * The RADIUS accounting 802.1X servers to be used for authentication. This param is only valid if the authMode is 'open-with-radius', '8021x-radius' or 'ipsk-with-radius' and radiusAccountingEnabled is 'true'
    * @member {Array.<module:model/NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers>} radiusAccountingServers
    */
    'radiusAccountingServers' = undefined;
    /**
    * Specify the RADIUS attribute used to look up group policies ('Filter-Id', 'Reply-Message', 'Airespace-ACL-Name' or 'Aruba-User-Role'). Access points must receive this attribute in the RADIUS Access-Accept message
    * @member {module:model/Body113.RadiusAttributeForGroupPoliciesEnum} radiusAttributeForGroupPolicies
    */
    'radiusAttributeForGroupPolicies' = undefined;
    /**
    * The client IP assignment mode ('NAT mode', 'Bridge mode', 'Layer 3 roaming', 'Layer 3 roaming with a concentrator' or 'VPN')
    * @member {String} ipAssignmentMode
    */
    'ipAssignmentMode' = undefined;
    /**
    * Whether or not traffic should be directed to use specific VLANs. This param is only valid if the ipAssignmentMode is 'Bridge mode' or 'Layer 3 roaming'
    * @member {Boolean} useVlanTagging
    */
    'useVlanTagging' = undefined;
    /**
    * The concentrator to use when the ipAssignmentMode is 'Layer 3 roaming with a concentrator' or 'VPN'.
    * @member {String} concentratorNetworkId
    */
    'concentratorNetworkId' = undefined;
    /**
    * The VLAN ID used for VLAN tagging. This param is only valid when the ipAssignmentMode is 'Layer 3 roaming with a concentrator' or 'VPN'
    * @member {Number} vlanId
    */
    'vlanId' = undefined;
    /**
    * The default VLAN ID used for 'all other APs'. This param is only valid when the ipAssignmentMode is 'Bridge mode' or 'Layer 3 roaming'
    * @member {Number} defaultVlanId
    */
    'defaultVlanId' = undefined;
    /**
    * The list of tags and VLAN IDs used for VLAN tagging. This param is only valid when the ipAssignmentMode is 'Bridge mode' or 'Layer 3 roaming'
    * @member {Array.<module:model/NetworksnetworkIdwirelessssidsnumberApTagsAndVlanIds>} apTagsAndVlanIds
    */
    'apTagsAndVlanIds' = undefined;
    /**
    * Allow access to a configurable list of IP ranges, which users may access prior to sign-on.
    * @member {Boolean} walledGardenEnabled
    */
    'walledGardenEnabled' = undefined;
    /**
    * Specify your walled garden by entering an array of addresses, ranges using CIDR notation, domain names, and domain wildcards (e.g. '192.168.1.1/24', '192.168.37.10/32', 'www.yahoo.com', '*.google.com']). Meraki's splash page is automatically included in your walled garden.
    * @member {Array.<String>} walledGardenRanges
    */
    'walledGardenRanges' = undefined;
    /**
    * If true, the RADIUS response can override VLAN tag. This is not valid when ipAssignmentMode is 'NAT mode'.
    * @member {Boolean} radiusOverride
    */
    'radiusOverride' = undefined;
    /**
    * Whether or not RADIUS Guest VLAN is enabled. This param is only valid if the authMode is 'open-with-radius' and addressing mode is not set to 'isolated' or 'nat' mode
    * @member {Boolean} radiusGuestVlanEnabled
    */
    'radiusGuestVlanEnabled' = undefined;
    /**
    * VLAN ID of the RADIUS Guest VLAN. This param is only valid if the authMode is 'open-with-radius' and addressing mode is not set to 'isolated' or 'nat' mode
    * @member {Number} radiusGuestVlanId
    */
    'radiusGuestVlanId' = undefined;
    /**
    * The minimum bitrate in Mbps. ('1', '2', '5.5', '6', '9', '11', '12', '18', '24', '36', '48' or '54')
    * @member {Number} minBitrate
    */
    'minBitrate' = undefined;
    /**
    * The client-serving radio frequencies. ('Dual band operation', '5 GHz band only' or 'Dual band operation with Band Steering')
    * @member {String} bandSelection
    */
    'bandSelection' = undefined;
    /**
    * The upload bandwidth limit in Kbps. (0 represents no limit.)
    * @member {Number} perClientBandwidthLimitUp
    */
    'perClientBandwidthLimitUp' = undefined;
    /**
    * The download bandwidth limit in Kbps. (0 represents no limit.)
    * @member {Number} perClientBandwidthLimitDown
    */
    'perClientBandwidthLimitDown' = undefined;
    /**
    * The total upload bandwidth limit in Kbps. (0 represents no limit.)
    * @member {Number} perSsidBandwidthLimitUp
    */
    'perSsidBandwidthLimitUp' = undefined;
    /**
    * The total download bandwidth limit in Kbps. (0 represents no limit.)
    * @member {Number} perSsidBandwidthLimitDown
    */
    'perSsidBandwidthLimitDown' = undefined;
    /**
    * Boolean indicating whether Layer 2 LAN isolation should be enabled or disabled. Only configurable when ipAssignmentMode is 'Bridge mode'.
    * @member {Boolean} lanIsolationEnabled
    */
    'lanIsolationEnabled' = undefined;
    /**
    * Boolean indicating whether APs should advertise or hide this SSID. APs will only broadcast this SSID if set to true
    * @member {Boolean} visible
    */
    'visible' = undefined;
    /**
    * Boolean indicating whether all APs should broadcast the SSID or if it should be restricted to APs matching any availability tags. Can only be false if the SSID has availability tags.
    * @member {Boolean} availableOnAllAps
    */
    'availableOnAllAps' = undefined;
    /**
    * Accepts a list of tags for this SSID. If availableOnAllAps is false, then the SSID will only be broadcast by APs with tags matching any of the tags in this list.
    * @member {Array.<String>} availabilityTags
    */
    'availabilityTags' = undefined;



    /**
    * Allowed values for the <code>authMode</code> property.
    * @enum {String}
    * @readonly
    */
    static AuthModeEnum = {
        /**
         * value: "open"
         * @const
         */
        "open": "open",
        /**
         * value: "psk"
         * @const
         */
        "psk": "psk",
        /**
         * value: "open-with-radius"
         * @const
         */
        "open-with-radius": "open-with-radius",
        /**
         * value: "8021x-meraki"
         * @const
         */
        "8021x-meraki": "8021x-meraki",
        /**
         * value: "8021x-radius"
         * @const
         */
        "8021x-radius": "8021x-radius",
        /**
         * value: "ipsk-with-radius"
         * @const
         */
        "ipsk-with-radius": "ipsk-with-radius",
        /**
         * value: "ipsk-without-radius"
         * @const
         */
        "ipsk-without-radius": "ipsk-without-radius"    };
    /**
    * Allowed values for the <code>enterpriseAdminAccess</code> property.
    * @enum {String}
    * @readonly
    */
    static EnterpriseAdminAccessEnum = {
        /**
         * value: "access disabled"
         * @const
         */
        "disabled": "access disabled",
        /**
         * value: "access enabled"
         * @const
         */
        "enabled": "access enabled"    };
    /**
    * Allowed values for the <code>encryptionMode</code> property.
    * @enum {String}
    * @readonly
    */
    static EncryptionModeEnum = {
        /**
         * value: "wep"
         * @const
         */
        "wep": "wep",
        /**
         * value: "wpa"
         * @const
         */
        "wpa": "wpa"    };
    /**
    * Allowed values for the <code>wpaEncryptionMode</code> property.
    * @enum {String}
    * @readonly
    */
    static WpaEncryptionModeEnum = {
        /**
         * value: "WPA1 only"
         * @const
         */
        "WPA1 only": "WPA1 only",
        /**
         * value: "WPA1 and WPA2"
         * @const
         */
        "WPA1 and WPA2": "WPA1 and WPA2",
        /**
         * value: "WPA2 only"
         * @const
         */
        "WPA2 only": "WPA2 only",
        /**
         * value: "WPA3 Transition Mode"
         * @const
         */
        "WPA3 Transition Mode": "WPA3 Transition Mode",
        /**
         * value: "WPA3 only"
         * @const
         */
        "WPA3 only": "WPA3 only"    };
    /**
    * Allowed values for the <code>splashPage</code> property.
    * @enum {String}
    * @readonly
    */
    static SplashPageEnum = {
        /**
         * value: "None"
         * @const
         */
        "None": "None",
        /**
         * value: "Click-through splash page"
         * @const
         */
        "Click-through splash page": "Click-through splash page",
        /**
         * value: "Billing"
         * @const
         */
        "Billing": "Billing",
        /**
         * value: "Password-protected with Meraki RADIUS"
         * @const
         */
        "Password-protected with Meraki RADIUS": "Password-protected with Meraki RADIUS",
        /**
         * value: "Password-protected with custom RADIUS"
         * @const
         */
        "Password-protected with custom RADIUS": "Password-protected with custom RADIUS",
        /**
         * value: "Password-protected with Active Directory"
         * @const
         */
        "Password-protected with Active Directory": "Password-protected with Active Directory",
        /**
         * value: "Password-protected with LDAP"
         * @const
         */
        "Password-protected with LDAP": "Password-protected with LDAP",
        /**
         * value: "SMS authentication"
         * @const
         */
        "SMS authentication": "SMS authentication",
        /**
         * value: "Systems Manager Sentry"
         * @const
         */
        "Systems Manager Sentry": "Systems Manager Sentry",
        /**
         * value: "Facebook Wi-Fi"
         * @const
         */
        "Facebook Wi-Fi": "Facebook Wi-Fi",
        /**
         * value: "Google OAuth"
         * @const
         */
        "Google OAuth": "Google OAuth",
        /**
         * value: "Sponsored guest"
         * @const
         */
        "Sponsored guest": "Sponsored guest"    };
    /**
    * Allowed values for the <code>radiusFailoverPolicy</code> property.
    * @enum {String}
    * @readonly
    */
    static RadiusFailoverPolicyEnum = {
        /**
         * value: "Deny access"
         * @const
         */
        "Deny access": "Deny access",
        /**
         * value: "Allow access"
         * @const
         */
        "Allow access": "Allow access"    };
    /**
    * Allowed values for the <code>radiusLoadBalancingPolicy</code> property.
    * @enum {String}
    * @readonly
    */
    static RadiusLoadBalancingPolicyEnum = {
        /**
         * value: "Strict priority order"
         * @const
         */
        "Strict priority order": "Strict priority order",
        /**
         * value: "Round robin"
         * @const
         */
        "Round robin": "Round robin"    };
    /**
    * Allowed values for the <code>radiusAttributeForGroupPolicies</code> property.
    * @enum {String}
    * @readonly
    */
    static RadiusAttributeForGroupPoliciesEnum = {
        /**
         * value: "Filter-Id"
         * @const
         */
        "Filter-Id": "Filter-Id",
        /**
         * value: "Reply-Message"
         * @const
         */
        "Reply-Message": "Reply-Message",
        /**
         * value: "Airespace-ACL-Name"
         * @const
         */
        "Airespace-ACL-Name": "Airespace-ACL-Name",
        /**
         * value: "Aruba-User-Role"
         * @const
         */
        "Aruba-User-Role": "Aruba-User-Role"    };

}
