// meraki_dashboard_api/src/model/Body4.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body4 model module.
* @module model/Body4
* @version 1.0.0
*/
export default class Body4 {
    /**
    * Constructs a new <code>Body4</code>.
    * @alias module:model/Body4
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body4</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body4} obj Optional instance to populate.
    * @return {module:model/Body4} The populated <code>Body4</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body4();
                        
            
            if (data.hasOwnProperty('senseEnabled')) {
                obj['senseEnabled'] = ApiClient.convertToType(data['senseEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('mqttBrokerId')) {
                obj['mqttBrokerId'] = ApiClient.convertToType(data['mqttBrokerId'], 'String');
            }
            if (data.hasOwnProperty('detectionModelId')) {
                obj['detectionModelId'] = ApiClient.convertToType(data['detectionModelId'], 'String');
            }
        }
        return obj;
    }

    /**
    * Boolean indicating if sense(license) is enabled(true) or disabled(false) on the camera
    * @member {Boolean} senseEnabled
    */
    'senseEnabled' = undefined;
    /**
    * The ID of the MQTT broker to be enabled on the camera. A value of null will disable MQTT on the camera
    * @member {String} mqttBrokerId
    */
    'mqttBrokerId' = undefined;
    /**
    * The ID of the object detection model
    * @member {String} detectionModelId
    */
    'detectionModelId' = undefined;




}
