// meraki_dashboard_api/src/model/Body44.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges from './NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges';
import NetworksnetworkIdappliancevlansvlanIdDhcpOptions from './NetworksnetworkIdappliancevlansvlanIdDhcpOptions';

/**
* The Body44 model module.
* @module model/Body44
* @version 1.0.0
*/
export default class Body44 {
    /**
    * Constructs a new <code>Body44</code>.
    * @alias module:model/Body44
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body44</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body44} obj Optional instance to populate.
    * @return {module:model/Body44} The populated <code>Body44</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body44();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('subnet')) {
                obj['subnet'] = ApiClient.convertToType(data['subnet'], 'String');
            }
            if (data.hasOwnProperty('applianceIp')) {
                obj['applianceIp'] = ApiClient.convertToType(data['applianceIp'], 'String');
            }
            if (data.hasOwnProperty('groupPolicyId')) {
                obj['groupPolicyId'] = ApiClient.convertToType(data['groupPolicyId'], 'String');
            }
            if (data.hasOwnProperty('vpnNatSubnet')) {
                obj['vpnNatSubnet'] = ApiClient.convertToType(data['vpnNatSubnet'], 'String');
            }
            if (data.hasOwnProperty('dhcpHandling')) {
                obj['dhcpHandling'] = ApiClient.convertToType(data['dhcpHandling'], 'String');
            }
            if (data.hasOwnProperty('dhcpRelayServerIps')) {
                obj['dhcpRelayServerIps'] = ApiClient.convertToType(data['dhcpRelayServerIps'], ['String']);
            }
            if (data.hasOwnProperty('dhcpLeaseTime')) {
                obj['dhcpLeaseTime'] = ApiClient.convertToType(data['dhcpLeaseTime'], 'String');
            }
            if (data.hasOwnProperty('dhcpBootOptionsEnabled')) {
                obj['dhcpBootOptionsEnabled'] = ApiClient.convertToType(data['dhcpBootOptionsEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('dhcpBootNextServer')) {
                obj['dhcpBootNextServer'] = ApiClient.convertToType(data['dhcpBootNextServer'], 'String');
            }
            if (data.hasOwnProperty('dhcpBootFilename')) {
                obj['dhcpBootFilename'] = ApiClient.convertToType(data['dhcpBootFilename'], 'String');
            }
            if (data.hasOwnProperty('fixedIpAssignments')) {
                obj['fixedIpAssignments'] = ApiClient.convertToType(data['fixedIpAssignments'], Object);
            }
            if (data.hasOwnProperty('reservedIpRanges')) {
                obj['reservedIpRanges'] = ApiClient.convertToType(data['reservedIpRanges'], [NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges]);
            }
            if (data.hasOwnProperty('dnsNameservers')) {
                obj['dnsNameservers'] = ApiClient.convertToType(data['dnsNameservers'], 'String');
            }
            if (data.hasOwnProperty('dhcpOptions')) {
                obj['dhcpOptions'] = ApiClient.convertToType(data['dhcpOptions'], [NetworksnetworkIdappliancevlansvlanIdDhcpOptions]);
            }
        }
        return obj;
    }

    /**
    * The name of the VLAN
    * @member {String} name
    */
    'name' = undefined;
    /**
    * The subnet of the VLAN
    * @member {String} subnet
    */
    'subnet' = undefined;
    /**
    * The local IP of the appliance on the VLAN
    * @member {String} applianceIp
    */
    'applianceIp' = undefined;
    /**
    * The id of the desired group policy to apply to the VLAN
    * @member {String} groupPolicyId
    */
    'groupPolicyId' = undefined;
    /**
    * The translated VPN subnet if VPN and VPN subnet translation are enabled on the VLAN
    * @member {String} vpnNatSubnet
    */
    'vpnNatSubnet' = undefined;
    /**
    * The appliance's handling of DHCP requests on this VLAN. One of: 'Run a DHCP server', 'Relay DHCP to another server' or 'Do not respond to DHCP requests'
    * @member {module:model/Body44.DhcpHandlingEnum} dhcpHandling
    */
    'dhcpHandling' = undefined;
    /**
    * The IPs of the DHCP servers that DHCP requests should be relayed to
    * @member {Array.<String>} dhcpRelayServerIps
    */
    'dhcpRelayServerIps' = undefined;
    /**
    * The term of DHCP leases if the appliance is running a DHCP server on this VLAN. One of: '30 minutes', '1 hour', '4 hours', '12 hours', '1 day' or '1 week'
    * @member {module:model/Body44.DhcpLeaseTimeEnum} dhcpLeaseTime
    */
    'dhcpLeaseTime' = undefined;
    /**
    * Use DHCP boot options specified in other properties
    * @member {Boolean} dhcpBootOptionsEnabled
    */
    'dhcpBootOptionsEnabled' = undefined;
    /**
    * DHCP boot option to direct boot clients to the server to load the boot file from
    * @member {String} dhcpBootNextServer
    */
    'dhcpBootNextServer' = undefined;
    /**
    * DHCP boot option for boot filename
    * @member {String} dhcpBootFilename
    */
    'dhcpBootFilename' = undefined;
    /**
    * The DHCP fixed IP assignments on the VLAN. This should be an object that contains mappings from MAC addresses to objects that themselves each contain \"ip\" and \"name\" string fields. See the sample request/response for more details.
    * @member {Object} fixedIpAssignments
    */
    'fixedIpAssignments' = undefined;
    /**
    * The DHCP reserved IP ranges on the VLAN
    * @member {Array.<module:model/NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges>} reservedIpRanges
    */
    'reservedIpRanges' = undefined;
    /**
    * The DNS nameservers used for DHCP responses, either \"upstream_dns\", \"google_dns\", \"opendns\", or a newline seperated string of IP addresses or domain names
    * @member {String} dnsNameservers
    */
    'dnsNameservers' = undefined;
    /**
    * The list of DHCP options that will be included in DHCP responses. Each object in the list should have \"code\", \"type\", and \"value\" properties.
    * @member {Array.<module:model/NetworksnetworkIdappliancevlansvlanIdDhcpOptions>} dhcpOptions
    */
    'dhcpOptions' = undefined;



    /**
    * Allowed values for the <code>dhcpHandling</code> property.
    * @enum {String}
    * @readonly
    */
    static DhcpHandlingEnum = {
        /**
         * value: "Run a DHCP server"
         * @const
         */
        "Run a DHCP server": "Run a DHCP server",
        /**
         * value: "Relay DHCP to another server"
         * @const
         */
        "Relay DHCP to another server": "Relay DHCP to another server",
        /**
         * value: "Do not respond to DHCP requests"
         * @const
         */
        "Do not respond to DHCP requests": "Do not respond to DHCP requests"    };
    /**
    * Allowed values for the <code>dhcpLeaseTime</code> property.
    * @enum {String}
    * @readonly
    */
    static DhcpLeaseTimeEnum = {
        /**
         * value: "30 minutes"
         * @const
         */
        "30 minutes": "30 minutes",
        /**
         * value: "1 hour"
         * @const
         */
        "1 hour": "1 hour",
        /**
         * value: "4 hours"
         * @const
         */
        "4 hours": "4 hours",
        /**
         * value: "12 hours"
         * @const
         */
        "12 hours": "12 hours",
        /**
         * value: "1 day"
         * @const
         */
        "1 day": "1 day",
        /**
         * value: "1 week"
         * @const
         */
        "1 week": "1 week"    };

}
