// meraki_dashboard_api/src/model/Body89.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body89 model module.
* @module model/Body89
* @version 1.0.0
*/
export default class Body89 {
    /**
    * Constructs a new <code>Body89</code>.
    * @alias module:model/Body89
    * @class
    * @param interfaceIp {String} The IP address of the interface where the RP needs to be created.
    * @param multicastGroup {String} 'Any', or the IP address of a multicast group
    */

    constructor(interfaceIp, multicastGroup) {
        
        
        this['interfaceIp'] = interfaceIp;
        this['multicastGroup'] = multicastGroup;
        
    }

    /**
    * Constructs a <code>Body89</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body89} obj Optional instance to populate.
    * @return {module:model/Body89} The populated <code>Body89</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body89();
                        
            
            if (data.hasOwnProperty('interfaceIp')) {
                obj['interfaceIp'] = ApiClient.convertToType(data['interfaceIp'], 'String');
            }
            if (data.hasOwnProperty('multicastGroup')) {
                obj['multicastGroup'] = ApiClient.convertToType(data['multicastGroup'], 'String');
            }
        }
        return obj;
    }

    /**
    * The IP address of the interface where the RP needs to be created.
    * @member {String} interfaceIp
    */
    'interfaceIp' = undefined;
    /**
    * 'Any', or the IP address of a multicast group
    * @member {String} multicastGroup
    */
    'multicastGroup' = undefined;




}
