// meraki_dashboard_api/src/model/Body62.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdgroupPoliciesBandwidth from './NetworksnetworkIdgroupPoliciesBandwidth';
import NetworksnetworkIdgroupPoliciesBonjourForwarding from './NetworksnetworkIdgroupPoliciesBonjourForwarding';
import NetworksnetworkIdgroupPoliciesContentFiltering from './NetworksnetworkIdgroupPoliciesContentFiltering';
import NetworksnetworkIdgroupPoliciesFirewallAndTrafficShaping from './NetworksnetworkIdgroupPoliciesFirewallAndTrafficShaping';
import NetworksnetworkIdgroupPoliciesScheduling from './NetworksnetworkIdgroupPoliciesScheduling';
import NetworksnetworkIdgroupPoliciesVlanTagging from './NetworksnetworkIdgroupPoliciesVlanTagging';

/**
* The Body62 model module.
* @module model/Body62
* @version 1.0.0
*/
export default class Body62 {
    /**
    * Constructs a new <code>Body62</code>.
    * @alias module:model/Body62
    * @class
    * @param name {String} The name for your group policy. Required.
    */

    constructor(name) {
        
        
        this['name'] = name;
        
    }

    /**
    * Constructs a <code>Body62</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body62} obj Optional instance to populate.
    * @return {module:model/Body62} The populated <code>Body62</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body62();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('scheduling')) {
                obj['scheduling'] = NetworksnetworkIdgroupPoliciesScheduling.constructFromObject(data['scheduling']);
            }
            if (data.hasOwnProperty('bandwidth')) {
                obj['bandwidth'] = NetworksnetworkIdgroupPoliciesBandwidth.constructFromObject(data['bandwidth']);
            }
            if (data.hasOwnProperty('firewallAndTrafficShaping')) {
                obj['firewallAndTrafficShaping'] = NetworksnetworkIdgroupPoliciesFirewallAndTrafficShaping.constructFromObject(data['firewallAndTrafficShaping']);
            }
            if (data.hasOwnProperty('contentFiltering')) {
                obj['contentFiltering'] = NetworksnetworkIdgroupPoliciesContentFiltering.constructFromObject(data['contentFiltering']);
            }
            if (data.hasOwnProperty('splashAuthSettings')) {
                obj['splashAuthSettings'] = ApiClient.convertToType(data['splashAuthSettings'], 'String');
            }
            if (data.hasOwnProperty('vlanTagging')) {
                obj['vlanTagging'] = NetworksnetworkIdgroupPoliciesVlanTagging.constructFromObject(data['vlanTagging']);
            }
            if (data.hasOwnProperty('bonjourForwarding')) {
                obj['bonjourForwarding'] = NetworksnetworkIdgroupPoliciesBonjourForwarding.constructFromObject(data['bonjourForwarding']);
            }
        }
        return obj;
    }

    /**
    * The name for your group policy. Required.
    * @member {String} name
    */
    'name' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesScheduling} scheduling
    */
    'scheduling' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesBandwidth} bandwidth
    */
    'bandwidth' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesFirewallAndTrafficShaping} firewallAndTrafficShaping
    */
    'firewallAndTrafficShaping' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesContentFiltering} contentFiltering
    */
    'contentFiltering' = undefined;
    /**
    * Whether clients bound to your policy will bypass splash authorization or behave according to the network's rules. Can be one of 'network default' or 'bypass'. Only available if your network has a wireless configuration.
    * @member {module:model/Body62.SplashAuthSettingsEnum} splashAuthSettings
    */
    'splashAuthSettings' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesVlanTagging} vlanTagging
    */
    'vlanTagging' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesBonjourForwarding} bonjourForwarding
    */
    'bonjourForwarding' = undefined;



    /**
    * Allowed values for the <code>splashAuthSettings</code> property.
    * @enum {String}
    * @readonly
    */
    static SplashAuthSettingsEnum = {
        /**
         * value: "network default"
         * @const
         */
        "network default": "network default",
        /**
         * value: "bypass"
         * @const
         */
        "bypass": "bypass"    };

}
