// meraki_dashboard_api/src/model/Body74.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body74 model module.
* @module model/Body74
* @version 1.0.0
*/
export default class Body74 {
    /**
    * Constructs a new <code>Body74</code>.
    * @alias module:model/Body74
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body74</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body74} obj Optional instance to populate.
    * @return {module:model/Body74} The populated <code>Body74</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body74();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('scope')) {
                obj['scope'] = ApiClient.convertToType(data['scope'], 'String');
            }
        }
        return obj;
    }

    /**
    * The name of this target group
    * @member {String} name
    */
    'name' = undefined;
    /**
    * The scope and tag options of the target group. Comma separated values beginning with one of withAny, withAll, withoutAny, withoutAll, all, none, followed by tags. Default to none if empty.
    * @member {String} scope
    */
    'scope' = undefined;




}
