// meraki_dashboard_api/src/model/Body108.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdwirelessalternateManagementInterfaceAccessPoints from './NetworksnetworkIdwirelessalternateManagementInterfaceAccessPoints';

/**
* The Body108 model module.
* @module model/Body108
* @version 1.0.0
*/
export default class Body108 {
    /**
    * Constructs a new <code>Body108</code>.
    * @alias module:model/Body108
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body108</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body108} obj Optional instance to populate.
    * @return {module:model/Body108} The populated <code>Body108</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body108();
                        
            
            if (data.hasOwnProperty('enabled')) {
                obj['enabled'] = ApiClient.convertToType(data['enabled'], 'Boolean');
            }
            if (data.hasOwnProperty('vlanId')) {
                obj['vlanId'] = ApiClient.convertToType(data['vlanId'], 'Number');
            }
            if (data.hasOwnProperty('protocols')) {
                obj['protocols'] = ApiClient.convertToType(data['protocols'], ['String']);
            }
            if (data.hasOwnProperty('accessPoints')) {
                obj['accessPoints'] = ApiClient.convertToType(data['accessPoints'], [NetworksnetworkIdwirelessalternateManagementInterfaceAccessPoints]);
            }
        }
        return obj;
    }

    /**
    * Boolean value to enable or disable alternate management interface
    * @member {Boolean} enabled
    */
    'enabled' = undefined;
    /**
    * Alternate management interface VLAN, must be between 1 and 4094
    * @member {Number} vlanId
    */
    'vlanId' = undefined;
    /**
    * Can be one or more of the following values: 'radius', 'snmp', 'syslog' or 'ldap'
    * @member {Array.<module:model/Body108.ProtocolsEnum>} protocols
    */
    'protocols' = undefined;
    /**
    * Array of access point serial number and IP assignment. Note: accessPoints IP assignment is not applicable for template networks, in other words, do not put 'accessPoints' in the body when updating template networks. Also, an empty 'accessPoints' array will remove all previous static IP assignments
    * @member {Array.<module:model/NetworksnetworkIdwirelessalternateManagementInterfaceAccessPoints>} accessPoints
    */
    'accessPoints' = undefined;



    /**
    * Allowed values for the <code>protocols</code> property.
    * @enum {String}
    * @readonly
    */
    static ProtocolsEnum = {
        /**
         * value: "radius"
         * @const
         */
        "radius": "radius",
        /**
         * value: "snmp"
         * @const
         */
        "snmp": "snmp",
        /**
         * value: "syslog"
         * @const
         */
        "syslog": "syslog",
        /**
         * value: "ldap"
         * @const
         */
        "ldap": "ldap"    };

}
