// meraki_dashboard_api/src/model/Body118.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body118 model module.
* @module model/Body118
* @version 1.0.0
*/
export default class Body118 {
    /**
    * Constructs a new <code>Body118</code>.
    * @alias module:model/Body118
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body118</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body118} obj Optional instance to populate.
    * @return {module:model/Body118} The populated <code>Body118</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body118();
                        
            
            if (data.hasOwnProperty('splashUrl')) {
                obj['splashUrl'] = ApiClient.convertToType(data['splashUrl'], 'String');
            }
            if (data.hasOwnProperty('useSplashUrl')) {
                obj['useSplashUrl'] = ApiClient.convertToType(data['useSplashUrl'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * [optional] The custom splash URL of the click-through splash page. Note that the URL can be configured without necessarily being used. In order to enable the custom URL, see 'useSplashUrl'
    * @member {String} splashUrl
    */
    'splashUrl' = undefined;
    /**
    * [optional] Boolean indicating whether the user will be redirected to the custom splash url. A custom splash URL must be set if this is true. Note that depending on your SSID's access control settings, it may not be possible to use the custom splash URL.
    * @member {Boolean} useSplashUrl
    */
    'useSplashUrl' = undefined;




}
