// meraki_dashboard_api/src/model/Body47.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body47 model module.
* @module model/Body47
* @version 1.0.0
*/
export default class Body47 {
    /**
    * Constructs a new <code>Body47</code>.
    * @alias module:model/Body47
    * @class
    * @param configTemplateId {String} The ID of the template to which the network should be bound.
    */

    constructor(configTemplateId) {
        
        
        this['configTemplateId'] = configTemplateId;
        
    }

    /**
    * Constructs a <code>Body47</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body47} obj Optional instance to populate.
    * @return {module:model/Body47} The populated <code>Body47</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body47();
                        
            
            if (data.hasOwnProperty('configTemplateId')) {
                obj['configTemplateId'] = ApiClient.convertToType(data['configTemplateId'], 'String');
            }
            if (data.hasOwnProperty('autoBind')) {
                obj['autoBind'] = ApiClient.convertToType(data['autoBind'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * The ID of the template to which the network should be bound.
    * @member {String} configTemplateId
    */
    'configTemplateId' = undefined;
    /**
    * Optional boolean indicating whether the network's switches should automatically bind to profiles of the same model. Defaults to false if left unspecified. This option only affects switch networks and switch templates. Auto-bind is not valid unless the switch template has at least one profile and has at most one profile per switch model.
    * @member {Boolean} autoBind
    */
    'autoBind' = undefined;




}
