// meraki_dashboard_api/src/model/Body25.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdappliancefirewallcellularFirewallRulesRules from './NetworksnetworkIdappliancefirewallcellularFirewallRulesRules';

/**
* The Body25 model module.
* @module model/Body25
* @version 1.0.0
*/
export default class Body25 {
    /**
    * Constructs a new <code>Body25</code>.
    * @alias module:model/Body25
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body25</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body25} obj Optional instance to populate.
    * @return {module:model/Body25} The populated <code>Body25</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body25();
                        
            
            if (data.hasOwnProperty('rules')) {
                obj['rules'] = ApiClient.convertToType(data['rules'], [NetworksnetworkIdappliancefirewallcellularFirewallRulesRules]);
            }
            if (data.hasOwnProperty('syslogDefaultRule')) {
                obj['syslogDefaultRule'] = ApiClient.convertToType(data['syslogDefaultRule'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * An ordered array of the firewall rules (not including the default rule)
    * @member {Array.<module:model/NetworksnetworkIdappliancefirewallcellularFirewallRulesRules>} rules
    */
    'rules' = undefined;
    /**
    * Log the special default rule (boolean value - enable only if you've configured a syslog server) (optional)
    * @member {Boolean} syslogDefaultRule
    */
    'syslogDefaultRule' = undefined;




}
