// meraki_dashboard_api/src/model/Body54.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdclientsprovisionClients from './NetworksnetworkIdclientsprovisionClients';
import NetworksnetworkIdclientsprovisionPoliciesBySecurityAppliance from './NetworksnetworkIdclientsprovisionPoliciesBySecurityAppliance';
import NetworksnetworkIdclientsprovisionPoliciesBySsid from './NetworksnetworkIdclientsprovisionPoliciesBySsid';

/**
* The Body54 model module.
* @module model/Body54
* @version 1.0.0
*/
export default class Body54 {
    /**
    * Constructs a new <code>Body54</code>.
    * @alias module:model/Body54
    * @class
    * @param clients {Array.<module:model/NetworksnetworkIdclientsprovisionClients>} The array of clients to provision
    * @param devicePolicy {module:model/Body54.DevicePolicyEnum} The policy to apply to the specified client. Can be 'Group policy', 'Allowed', 'Blocked', 'Per connection' or 'Normal'. Required.
    */

    constructor(clients, devicePolicy) {
        
        
        this['clients'] = clients;
        this['devicePolicy'] = devicePolicy;
        
    }

    /**
    * Constructs a <code>Body54</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body54} obj Optional instance to populate.
    * @return {module:model/Body54} The populated <code>Body54</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body54();
                        
            
            if (data.hasOwnProperty('clients')) {
                obj['clients'] = ApiClient.convertToType(data['clients'], [NetworksnetworkIdclientsprovisionClients]);
            }
            if (data.hasOwnProperty('devicePolicy')) {
                obj['devicePolicy'] = ApiClient.convertToType(data['devicePolicy'], 'String');
            }
            if (data.hasOwnProperty('groupPolicyId')) {
                obj['groupPolicyId'] = ApiClient.convertToType(data['groupPolicyId'], 'String');
            }
            if (data.hasOwnProperty('policiesBySecurityAppliance')) {
                obj['policiesBySecurityAppliance'] = NetworksnetworkIdclientsprovisionPoliciesBySecurityAppliance.constructFromObject(data['policiesBySecurityAppliance']);
            }
            if (data.hasOwnProperty('policiesBySsid')) {
                obj['policiesBySsid'] = NetworksnetworkIdclientsprovisionPoliciesBySsid.constructFromObject(data['policiesBySsid']);
            }
        }
        return obj;
    }

    /**
    * The array of clients to provision
    * @member {Array.<module:model/NetworksnetworkIdclientsprovisionClients>} clients
    */
    'clients' = undefined;
    /**
    * The policy to apply to the specified client. Can be 'Group policy', 'Allowed', 'Blocked', 'Per connection' or 'Normal'. Required.
    * @member {module:model/Body54.DevicePolicyEnum} devicePolicy
    */
    'devicePolicy' = undefined;
    /**
    * The ID of the desired group policy to apply to the client. Required if 'devicePolicy' is set to \"Group policy\". Otherwise this is ignored.
    * @member {String} groupPolicyId
    */
    'groupPolicyId' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdclientsprovisionPoliciesBySecurityAppliance} policiesBySecurityAppliance
    */
    'policiesBySecurityAppliance' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdclientsprovisionPoliciesBySsid} policiesBySsid
    */
    'policiesBySsid' = undefined;



    /**
    * Allowed values for the <code>devicePolicy</code> property.
    * @enum {String}
    * @readonly
    */
    static DevicePolicyEnum = {
        /**
         * value: "Group policy"
         * @const
         */
        "Group policy": "Group policy",
        /**
         * value: "Allowed"
         * @const
         */
        "Allowed": "Allowed",
        /**
         * value: "Blocked"
         * @const
         */
        "Blocked": "Blocked",
        /**
         * value: "Per connection"
         * @const
         */
        "Per connection": "Per connection",
        /**
         * value: "Normal"
         * @const
         */
        "Normal": "Normal"    };

}
