// meraki_dashboard_api/src/model/Body46.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body46 model module.
* @module model/Body46
* @version 1.0.0
*/
export default class Body46 {
    /**
    * Constructs a new <code>Body46</code>.
    * @alias module:model/Body46
    * @class
    * @param enabled {Boolean} Enable warm spare
    */

    constructor(enabled) {
        
        
        this['enabled'] = enabled;
        
    }

    /**
    * Constructs a <code>Body46</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body46} obj Optional instance to populate.
    * @return {module:model/Body46} The populated <code>Body46</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body46();
                        
            
            if (data.hasOwnProperty('enabled')) {
                obj['enabled'] = ApiClient.convertToType(data['enabled'], 'Boolean');
            }
            if (data.hasOwnProperty('spareSerial')) {
                obj['spareSerial'] = ApiClient.convertToType(data['spareSerial'], 'String');
            }
            if (data.hasOwnProperty('uplinkMode')) {
                obj['uplinkMode'] = ApiClient.convertToType(data['uplinkMode'], 'String');
            }
            if (data.hasOwnProperty('virtualIp1')) {
                obj['virtualIp1'] = ApiClient.convertToType(data['virtualIp1'], 'String');
            }
            if (data.hasOwnProperty('virtualIp2')) {
                obj['virtualIp2'] = ApiClient.convertToType(data['virtualIp2'], 'String');
            }
        }
        return obj;
    }

    /**
    * Enable warm spare
    * @member {Boolean} enabled
    */
    'enabled' = undefined;
    /**
    * Serial number of the warm spare appliance
    * @member {String} spareSerial
    */
    'spareSerial' = undefined;
    /**
    * Uplink mode, either virtual or public
    * @member {String} uplinkMode
    */
    'uplinkMode' = undefined;
    /**
    * The WAN 1 shared IP
    * @member {String} virtualIp1
    */
    'virtualIp1' = undefined;
    /**
    * The WAN 2 shared IP
    * @member {String} virtualIp2
    */
    'virtualIp2' = undefined;




}
