// meraki_dashboard_api/src/model/Body106.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body106 model module.
* @module model/Body106
* @version 1.0.0
*/
export default class Body106 {
    /**
    * Constructs a new <code>Body106</code>.
    * @alias module:model/Body106
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body106</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body106} obj Optional instance to populate.
    * @return {module:model/Body106} The populated <code>Body106</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body106();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('url')) {
                obj['url'] = ApiClient.convertToType(data['url'], 'String');
            }
            if (data.hasOwnProperty('sharedSecret')) {
                obj['sharedSecret'] = ApiClient.convertToType(data['sharedSecret'], 'String');
            }
        }
        return obj;
    }

    /**
    * A name for easy reference to the HTTP server
    * @member {String} name
    */
    'name' = undefined;
    /**
    * The URL of the HTTP server
    * @member {String} url
    */
    'url' = undefined;
    /**
    * A shared secret that will be included in POSTs sent to the HTTP server. This secret can be used to verify that the request was sent by Meraki.
    * @member {String} sharedSecret
    */
    'sharedSecret' = undefined;




}
