// meraki_dashboard_api/src/model/Body104.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdtrafficAnalysisCustomPieChartItems from './NetworksnetworkIdtrafficAnalysisCustomPieChartItems';

/**
* The Body104 model module.
* @module model/Body104
* @version 1.0.0
*/
export default class Body104 {
    /**
    * Constructs a new <code>Body104</code>.
    * @alias module:model/Body104
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body104</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body104} obj Optional instance to populate.
    * @return {module:model/Body104} The populated <code>Body104</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body104();
                        
            
            if (data.hasOwnProperty('mode')) {
                obj['mode'] = ApiClient.convertToType(data['mode'], 'String');
            }
            if (data.hasOwnProperty('customPieChartItems')) {
                obj['customPieChartItems'] = ApiClient.convertToType(data['customPieChartItems'], [NetworksnetworkIdtrafficAnalysisCustomPieChartItems]);
            }
        }
        return obj;
    }

    /**
    *     The traffic analysis mode for the network. Can be one of 'disabled' (do not collect traffic types),     'basic' (collect generic traffic categories), or 'detailed' (collect destination hostnames). 
    * @member {module:model/Body104.ModeEnum} mode
    */
    'mode' = undefined;
    /**
    * The list of items that make up the custom pie chart for traffic reporting.
    * @member {Array.<module:model/NetworksnetworkIdtrafficAnalysisCustomPieChartItems>} customPieChartItems
    */
    'customPieChartItems' = undefined;



    /**
    * Allowed values for the <code>mode</code> property.
    * @enum {String}
    * @readonly
    */
    static ModeEnum = {
        /**
         * value: "disabled"
         * @const
         */
        "disabled": "disabled",
        /**
         * value: "basic"
         * @const
         */
        "basic": "basic",
        /**
         * value: "detailed"
         * @const
         */
        "detailed": "detailed"    };

}
