// meraki_dashboard_api/src/model/Body78.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body78 model module.
* @module model/Body78
* @version 1.0.0
*/
export default class Body78 {
    /**
    * Constructs a new <code>Body78</code>.
    * @alias module:model/Body78
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body78</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body78} obj Optional instance to populate.
    * @return {module:model/Body78} The populated <code>Body78</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body78();
                        
            
            if (data.hasOwnProperty('defaultPolicy')) {
                obj['defaultPolicy'] = ApiClient.convertToType(data['defaultPolicy'], 'String');
            }
            if (data.hasOwnProperty('allowedServers')) {
                obj['allowedServers'] = ApiClient.convertToType(data['allowedServers'], ['String']);
            }
            if (data.hasOwnProperty('blockedServers')) {
                obj['blockedServers'] = ApiClient.convertToType(data['blockedServers'], ['String']);
            }
        }
        return obj;
    }

    /**
    * 'allow' or 'block' new DHCP servers. Default value is 'allow'.
    * @member {module:model/Body78.DefaultPolicyEnum} defaultPolicy
    */
    'defaultPolicy' = undefined;
    /**
    * List the MAC addresses of DHCP servers to permit on the network. Applicable only if defaultPolicy is set to block. An empty array will clear the entries.
    * @member {Array.<String>} allowedServers
    */
    'allowedServers' = undefined;
    /**
    * List the MAC addresses of DHCP servers to block on the network. Applicable only if defaultPolicy is set to allow. An empty array will clear the entries.
    * @member {Array.<String>} blockedServers
    */
    'blockedServers' = undefined;



    /**
    * Allowed values for the <code>defaultPolicy</code> property.
    * @enum {String}
    * @readonly
    */
    static DefaultPolicyEnum = {
        /**
         * value: "allow"
         * @const
         */
        "allow": "allow",
        /**
         * value: "block"
         * @const
         */
        "block": "block"    };

}
