// meraki_dashboard_api/src/model/Body107.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body107 model module.
* @module model/Body107
* @version 1.0.0
*/
export default class Body107 {
    /**
    * Constructs a new <code>Body107</code>.
    * @alias module:model/Body107
    * @class
    * @param url {String} The URL where the test webhook will be sent
    */

    constructor(url) {
        
        
        this['url'] = url;
        
    }

    /**
    * Constructs a <code>Body107</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body107} obj Optional instance to populate.
    * @return {module:model/Body107} The populated <code>Body107</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body107();
                        
            
            if (data.hasOwnProperty('url')) {
                obj['url'] = ApiClient.convertToType(data['url'], 'String');
            }
            if (data.hasOwnProperty('sharedSecret')) {
                obj['sharedSecret'] = ApiClient.convertToType(data['sharedSecret'], 'String');
            }
        }
        return obj;
    }

    /**
    * The URL where the test webhook will be sent
    * @member {String} url
    */
    'url' = undefined;
    /**
    * The shared secret the test webhook will send. Optional. Defaults to an empty string.
    * @member {String} sharedSecret
    * @default ''
    */
    'sharedSecret' = '';




}
