// meraki_dashboard_api/src/model/Body67.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body67 model module.
* @module model/Body67
* @version 1.0.0
*/
export default class Body67 {
    /**
    * Constructs a new <code>Body67</code>.
    * @alias module:model/Body67
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body67</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body67} obj Optional instance to populate.
    * @return {module:model/Body67} The populated <code>Body67</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body67();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('host')) {
                obj['host'] = ApiClient.convertToType(data['host'], 'String');
            }
            if (data.hasOwnProperty('port')) {
                obj['port'] = ApiClient.convertToType(data['port'], 'Number');
            }
        }
        return obj;
    }

    /**
    * Name of the mqtt config
    * @member {String} name
    */
    'name' = undefined;
    /**
    * Host name where mqtt broker runs
    * @member {String} host
    */
    'host' = undefined;
    /**
    * Host port though which mqtt broker can be reached
    * @member {Number} port
    */
    'port' = undefined;




}
