// meraki_dashboard_api/src/model/Body125.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import OrganizationsorganizationIdadminsNetworks from './OrganizationsorganizationIdadminsNetworks';
import OrganizationsorganizationIdadminsTags from './OrganizationsorganizationIdadminsTags';

/**
* The Body125 model module.
* @module model/Body125
* @version 1.0.0
*/
export default class Body125 {
    /**
    * Constructs a new <code>Body125</code>.
    * @alias module:model/Body125
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body125</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body125} obj Optional instance to populate.
    * @return {module:model/Body125} The populated <code>Body125</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body125();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('orgAccess')) {
                obj['orgAccess'] = ApiClient.convertToType(data['orgAccess'], 'String');
            }
            if (data.hasOwnProperty('tags')) {
                obj['tags'] = ApiClient.convertToType(data['tags'], [OrganizationsorganizationIdadminsTags]);
            }
            if (data.hasOwnProperty('networks')) {
                obj['networks'] = ApiClient.convertToType(data['networks'], [OrganizationsorganizationIdadminsNetworks]);
            }
        }
        return obj;
    }

    /**
    * The name of the dashboard administrator
    * @member {String} name
    */
    'name' = undefined;
    /**
    * The privilege of the dashboard administrator on the organization. Can be one of 'full', 'read-only', 'enterprise' or 'none'
    * @member {module:model/Body125.OrgAccessEnum} orgAccess
    */
    'orgAccess' = undefined;
    /**
    * The list of tags that the dashboard administrator has privileges on
    * @member {Array.<module:model/OrganizationsorganizationIdadminsTags>} tags
    */
    'tags' = undefined;
    /**
    * The list of networks that the dashboard administrator has privileges on
    * @member {Array.<module:model/OrganizationsorganizationIdadminsNetworks>} networks
    */
    'networks' = undefined;



    /**
    * Allowed values for the <code>orgAccess</code> property.
    * @enum {String}
    * @readonly
    */
    static OrgAccessEnum = {
        /**
         * value: "full"
         * @const
         */
        "full": "full",
        /**
         * value: "read-only"
         * @const
         */
        "read-only": "read-only",
        /**
         * value: "enterprise"
         * @const
         */
        "enterprise": "enterprise",
        /**
         * value: "none"
         * @const
         */
        "none": "none"    };

}
