// meraki_dashboard_api/src/model/Body137.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The Body137 model module.
* @module model/Body137
* @version 1.0.0
*/
export default class Body137 {
    /**
    * Constructs a new <code>Body137</code>.
    * @alias module:model/Body137
    * @class
    * @param name {String} The name of the VoIP provider
    * @param address {String} The IP address (IPv4 only) or hostname of the media server to monitor
    */

    constructor(name, address) {
        
        
        this['name'] = name;
        this['address'] = address;
        
    }

    /**
    * Constructs a <code>Body137</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body137} obj Optional instance to populate.
    * @return {module:model/Body137} The populated <code>Body137</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body137();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('address')) {
                obj['address'] = ApiClient.convertToType(data['address'], 'String');
            }
        }
        return obj;
    }

    /**
    * The name of the VoIP provider
    * @member {String} name
    */
    'name' = undefined;
    /**
    * The IP address (IPv4 only) or hostname of the media server to monitor
    * @member {String} address
    */
    'address' = undefined;




}
