// meraki_dashboard_api/src/model/Body114.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdwirelessssidsnumberfirewalll3FirewallRulesRules from './NetworksnetworkIdwirelessssidsnumberfirewalll3FirewallRulesRules';

/**
* The Body114 model module.
* @module model/Body114
* @version 1.0.0
*/
export default class Body114 {
    /**
    * Constructs a new <code>Body114</code>.
    * @alias module:model/Body114
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>Body114</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Body114} obj Optional instance to populate.
    * @return {module:model/Body114} The populated <code>Body114</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new Body114();
                        
            
            if (data.hasOwnProperty('rules')) {
                obj['rules'] = ApiClient.convertToType(data['rules'], [NetworksnetworkIdwirelessssidsnumberfirewalll3FirewallRulesRules]);
            }
            if (data.hasOwnProperty('allowLanAccess')) {
                obj['allowLanAccess'] = ApiClient.convertToType(data['allowLanAccess'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * An ordered array of the firewall rules for this SSID (not including the local LAN access rule or the default rule)
    * @member {Array.<module:model/NetworksnetworkIdwirelessssidsnumberfirewalll3FirewallRulesRules>} rules
    */
    'rules' = undefined;
    /**
    * Allow wireless client access to local LAN (boolean value - true allows access and false denies access) (optional)
    * @member {Boolean} allowLanAccess
    */
    'allowLanAccess' = undefined;




}
