// meraki_dashboard_api/src/model/DevicesserialcellularGatewaylanFixedIpAssignments.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The DevicesserialcellularGatewaylanFixedIpAssignments model module.
* @module model/DevicesserialcellularGatewaylanFixedIpAssignments
* @version 1.0.0
*/
export default class DevicesserialcellularGatewaylanFixedIpAssignments {
    /**
    * Constructs a new <code>DevicesserialcellularGatewaylanFixedIpAssignments</code>.
    * @alias module:model/DevicesserialcellularGatewaylanFixedIpAssignments
    * @class
    * @param ip {String} The IP address you want to assign to a specific server or device
    * @param mac {String} The MAC address of the server or device that hosts the internal resource that you wish to receive the specified IP address
    */

    constructor(ip, mac) {
        
        
        this['ip'] = ip;
        this['mac'] = mac;
        
    }

    /**
    * Constructs a <code>DevicesserialcellularGatewaylanFixedIpAssignments</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/DevicesserialcellularGatewaylanFixedIpAssignments} obj Optional instance to populate.
    * @return {module:model/DevicesserialcellularGatewaylanFixedIpAssignments} The populated <code>DevicesserialcellularGatewaylanFixedIpAssignments</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new DevicesserialcellularGatewaylanFixedIpAssignments();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('ip')) {
                obj['ip'] = ApiClient.convertToType(data['ip'], 'String');
            }
            if (data.hasOwnProperty('mac')) {
                obj['mac'] = ApiClient.convertToType(data['mac'], 'String');
            }
        }
        return obj;
    }

    /**
    * A descriptive name of the assignment
    * @member {String} name
    */
    'name' = undefined;
    /**
    * The IP address you want to assign to a specific server or device
    * @member {String} ip
    */
    'ip' = undefined;
    /**
    * The MAC address of the server or device that hosts the internal resource that you wish to receive the specified IP address
    * @member {String} mac
    */
    'mac' = undefined;




}
