// meraki_dashboard_api/src/model/NetworksnetworkIdswitchroutingospfMd5AuthenticationKey.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdswitchroutingospfMd5AuthenticationKey model module.
* @module model/NetworksnetworkIdswitchroutingospfMd5AuthenticationKey
* @version 1.0.0
*/
export default class NetworksnetworkIdswitchroutingospfMd5AuthenticationKey {
    /**
    * Constructs a new <code>NetworksnetworkIdswitchroutingospfMd5AuthenticationKey</code>.
    * MD5 authentication credentials. This param is only relevant if md5AuthenticationEnabled is true
    * @alias module:model/NetworksnetworkIdswitchroutingospfMd5AuthenticationKey
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdswitchroutingospfMd5AuthenticationKey</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdswitchroutingospfMd5AuthenticationKey} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdswitchroutingospfMd5AuthenticationKey} The populated <code>NetworksnetworkIdswitchroutingospfMd5AuthenticationKey</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdswitchroutingospfMd5AuthenticationKey();
                        
            
            if (data.hasOwnProperty('id')) {
                obj['id'] = ApiClient.convertToType(data['id'], 'Number');
            }
            if (data.hasOwnProperty('passphrase')) {
                obj['passphrase'] = ApiClient.convertToType(data['passphrase'], 'String');
            }
        }
        return obj;
    }

    /**
    * MD5 authentication key index. Key index must be between 1 to 255
    * @member {Number} id
    */
    'id' = undefined;
    /**
    * MD5 authentication passphrase
    * @member {String} passphrase
    */
    'passphrase' = undefined;




}
