// meraki_dashboard_api/src/model/NetworksnetworkIdsmdevicesfieldsDeviceFields.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdsmdevicesfieldsDeviceFields model module.
* @module model/NetworksnetworkIdsmdevicesfieldsDeviceFields
* @version 1.0.0
*/
export default class NetworksnetworkIdsmdevicesfieldsDeviceFields {
    /**
    * Constructs a new <code>NetworksnetworkIdsmdevicesfieldsDeviceFields</code>.
    * The new fields of the device. Each field of this object is optional.
    * @alias module:model/NetworksnetworkIdsmdevicesfieldsDeviceFields
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdsmdevicesfieldsDeviceFields</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdsmdevicesfieldsDeviceFields} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdsmdevicesfieldsDeviceFields} The populated <code>NetworksnetworkIdsmdevicesfieldsDeviceFields</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdsmdevicesfieldsDeviceFields();
                        
            
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('notes')) {
                obj['notes'] = ApiClient.convertToType(data['notes'], 'String');
            }
        }
        return obj;
    }

    /**
    * New name for the device
    * @member {String} name
    */
    'name' = undefined;
    /**
    * New notes for the device
    * @member {String} notes
    */
    'notes' = undefined;




}
