// meraki_dashboard_api/src/model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X model module.
* @module model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X
* @version 1.0.0
*/
export default class NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X {
    /**
    * Constructs a new <code>NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X</code>.
    * Quality and resolution for MV22X/MV72X camera models.
    * @alias module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X
    * @class
    * @param quality {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X.QualityEnum} Quality of the camera. Can be one of 'Standard', 'Enhanced' or 'High'.
    * @param resolution {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X.ResolutionEnum} Resolution of the camera. Can be one of '1280x720', '1920x1080' or '2688x1512'.
    */

    constructor(quality, resolution) {
        
        
        this['quality'] = quality;
        this['resolution'] = resolution;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X} The populated <code>NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X();
                        
            
            if (data.hasOwnProperty('quality')) {
                obj['quality'] = ApiClient.convertToType(data['quality'], 'String');
            }
            if (data.hasOwnProperty('resolution')) {
                obj['resolution'] = ApiClient.convertToType(data['resolution'], 'String');
            }
        }
        return obj;
    }

    /**
    * Quality of the camera. Can be one of 'Standard', 'Enhanced' or 'High'.
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X.QualityEnum} quality
    */
    'quality' = undefined;
    /**
    * Resolution of the camera. Can be one of '1280x720', '1920x1080' or '2688x1512'.
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X.ResolutionEnum} resolution
    */
    'resolution' = undefined;



    /**
    * Allowed values for the <code>quality</code> property.
    * @enum {String}
    * @readonly
    */
    static QualityEnum = {
        /**
         * value: "Standard"
         * @const
         */
        "Standard": "Standard",
        /**
         * value: "Enhanced"
         * @const
         */
        "Enhanced": "Enhanced",
        /**
         * value: "High"
         * @const
         */
        "High": "High"    };
    /**
    * Allowed values for the <code>resolution</code> property.
    * @enum {String}
    * @readonly
    */
    static ResolutionEnum = {
        /**
         * value: "1280x720"
         * @const
         */
        "1280x720": "1280x720",
        /**
         * value: "1920x1080"
         * @const
         */
        "1920x1080": "1920x1080",
        /**
         * value: "2688x1512"
         * @const
         */
        "2688x1512": "2688x1512"    };

}
