// meraki_dashboard_api/src/model/NetworksnetworkIdsyslogServersServers.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdsyslogServersServers model module.
* @module model/NetworksnetworkIdsyslogServersServers
* @version 1.0.0
*/
export default class NetworksnetworkIdsyslogServersServers {
    /**
    * Constructs a new <code>NetworksnetworkIdsyslogServersServers</code>.
    * @alias module:model/NetworksnetworkIdsyslogServersServers
    * @class
    * @param host {String} The IP address of the syslog server
    * @param port {Number} The port of the syslog server
    * @param roles {Array.<String>} A list of roles for the syslog server. Options (case-insensitive): 'Wireless event log', 'Appliance event log', 'Switch event log', 'Air Marshal events', 'Flows', 'URLs', 'IDS alerts', 'Security events'
    */

    constructor(host, port, roles) {
        
        
        this['host'] = host;
        this['port'] = port;
        this['roles'] = roles;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdsyslogServersServers</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdsyslogServersServers} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdsyslogServersServers} The populated <code>NetworksnetworkIdsyslogServersServers</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdsyslogServersServers();
                        
            
            if (data.hasOwnProperty('host')) {
                obj['host'] = ApiClient.convertToType(data['host'], 'String');
            }
            if (data.hasOwnProperty('port')) {
                obj['port'] = ApiClient.convertToType(data['port'], 'Number');
            }
            if (data.hasOwnProperty('roles')) {
                obj['roles'] = ApiClient.convertToType(data['roles'], ['String']);
            }
        }
        return obj;
    }

    /**
    * The IP address of the syslog server
    * @member {String} host
    */
    'host' = undefined;
    /**
    * The port of the syslog server
    * @member {Number} port
    */
    'port' = undefined;
    /**
    * A list of roles for the syslog server. Options (case-insensitive): 'Wireless event log', 'Appliance event log', 'Switch event log', 'Air Marshal events', 'Flows', 'URLs', 'IDS alerts', 'Security events'
    * @member {Array.<String>} roles
    */
    'roles' = undefined;




}
