// meraki_dashboard_api/src/model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdappliancetrafficShapinguplinkSelectionPerformanceClass from './NetworksnetworkIdappliancetrafficShapinguplinkSelectionPerformanceClass';
import NetworksnetworkIdappliancetrafficShapinguplinkSelectionTrafficFilters1 from './NetworksnetworkIdappliancetrafficShapinguplinkSelectionTrafficFilters1';

/**
* The NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences model module.
* @module model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences
* @version 1.0.0
*/
export default class NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences {
    /**
    * Constructs a new <code>NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences</code>.
    * @alias module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences
    * @class
    * @param trafficFilters {Array.<module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionTrafficFilters1>} Array of traffic filters for this uplink preference rule
    * @param preferredUplink {module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences.PreferredUplinkEnum} Preferred uplink for this uplink preference rule. Must be one of: 'wan1', 'wan2', 'bestForVoIP', 'loadBalancing' or 'defaultUplink'
    */

    constructor(trafficFilters, preferredUplink) {
        
        
        this['trafficFilters'] = trafficFilters;
        this['preferredUplink'] = preferredUplink;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences} The populated <code>NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences();
                        
            
            if (data.hasOwnProperty('trafficFilters')) {
                obj['trafficFilters'] = ApiClient.convertToType(data['trafficFilters'], [NetworksnetworkIdappliancetrafficShapinguplinkSelectionTrafficFilters1]);
            }
            if (data.hasOwnProperty('preferredUplink')) {
                obj['preferredUplink'] = ApiClient.convertToType(data['preferredUplink'], 'String');
            }
            if (data.hasOwnProperty('failOverCriterion')) {
                obj['failOverCriterion'] = ApiClient.convertToType(data['failOverCriterion'], 'String');
            }
            if (data.hasOwnProperty('performanceClass')) {
                obj['performanceClass'] = NetworksnetworkIdappliancetrafficShapinguplinkSelectionPerformanceClass.constructFromObject(data['performanceClass']);
            }
        }
        return obj;
    }

    /**
    * Array of traffic filters for this uplink preference rule
    * @member {Array.<module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionTrafficFilters1>} trafficFilters
    */
    'trafficFilters' = undefined;
    /**
    * Preferred uplink for this uplink preference rule. Must be one of: 'wan1', 'wan2', 'bestForVoIP', 'loadBalancing' or 'defaultUplink'
    * @member {module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences.PreferredUplinkEnum} preferredUplink
    */
    'preferredUplink' = undefined;
    /**
    * Fail over criterion for this uplink preference rule. Must be one of: 'poorPerformance' or 'uplinkDown'
    * @member {module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionVpnTrafficUplinkPreferences.FailOverCriterionEnum} failOverCriterion
    */
    'failOverCriterion' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdappliancetrafficShapinguplinkSelectionPerformanceClass} performanceClass
    */
    'performanceClass' = undefined;



    /**
    * Allowed values for the <code>preferredUplink</code> property.
    * @enum {String}
    * @readonly
    */
    static PreferredUplinkEnum = {
        /**
         * value: "wan1"
         * @const
         */
        "wan1": "wan1",
        /**
         * value: "wan2"
         * @const
         */
        "wan2": "wan2",
        /**
         * value: "bestForVoIP"
         * @const
         */
        "bestForVoIP": "bestForVoIP",
        /**
         * value: "loadBalancing"
         * @const
         */
        "loadBalancing": "loadBalancing",
        /**
         * value: "defaultUplink"
         * @const
         */
        "defaultUplink": "defaultUplink"    };
    /**
    * Allowed values for the <code>failOverCriterion</code> property.
    * @enum {String}
    * @readonly
    */
    static FailOverCriterionEnum = {
        /**
         * value: "poorPerformance"
         * @const
         */
        "poorPerformance": "poorPerformance",
        /**
         * value: "uplinkDown"
         * @const
         */
        "uplinkDown": "uplinkDown"    };

}
