// meraki_dashboard_api/src/model/NetworksnetworkIdswitchlinkAggregationsSwitchPorts.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdswitchlinkAggregationsSwitchPorts model module.
* @module model/NetworksnetworkIdswitchlinkAggregationsSwitchPorts
* @version 1.0.0
*/
export default class NetworksnetworkIdswitchlinkAggregationsSwitchPorts {
    /**
    * Constructs a new <code>NetworksnetworkIdswitchlinkAggregationsSwitchPorts</code>.
    * @alias module:model/NetworksnetworkIdswitchlinkAggregationsSwitchPorts
    * @class
    * @param serial {String} Serial number of the switch.
    * @param portId {String} Port identifier of switch port. For modules, the identifier is \"SlotNumber_ModuleType_PortNumber\" (Ex: “1_8X10G_1”), otherwise it is just the port number (Ex: \"8\").
    */

    constructor(serial, portId) {
        
        
        this['serial'] = serial;
        this['portId'] = portId;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdswitchlinkAggregationsSwitchPorts</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdswitchlinkAggregationsSwitchPorts} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdswitchlinkAggregationsSwitchPorts} The populated <code>NetworksnetworkIdswitchlinkAggregationsSwitchPorts</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdswitchlinkAggregationsSwitchPorts();
                        
            
            if (data.hasOwnProperty('serial')) {
                obj['serial'] = ApiClient.convertToType(data['serial'], 'String');
            }
            if (data.hasOwnProperty('portId')) {
                obj['portId'] = ApiClient.convertToType(data['portId'], 'String');
            }
        }
        return obj;
    }

    /**
    * Serial number of the switch.
    * @member {String} serial
    */
    'serial' = undefined;
    /**
    * Port identifier of switch port. For modules, the identifier is \"SlotNumber_ModuleType_PortNumber\" (Ex: “1_8X10G_1”), otherwise it is just the port number (Ex: \"8\").
    * @member {String} portId
    */
    'portId' = undefined;




}
