// meraki_dashboard_api/src/model/NetworksnetworkIdgroupPoliciesScheduling.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdgroupPoliciesSchedulingFriday from './NetworksnetworkIdgroupPoliciesSchedulingFriday';
import NetworksnetworkIdgroupPoliciesSchedulingMonday from './NetworksnetworkIdgroupPoliciesSchedulingMonday';
import NetworksnetworkIdgroupPoliciesSchedulingSaturday from './NetworksnetworkIdgroupPoliciesSchedulingSaturday';
import NetworksnetworkIdgroupPoliciesSchedulingSunday from './NetworksnetworkIdgroupPoliciesSchedulingSunday';
import NetworksnetworkIdgroupPoliciesSchedulingThursday from './NetworksnetworkIdgroupPoliciesSchedulingThursday';
import NetworksnetworkIdgroupPoliciesSchedulingTuesday from './NetworksnetworkIdgroupPoliciesSchedulingTuesday';
import NetworksnetworkIdgroupPoliciesSchedulingWednesday from './NetworksnetworkIdgroupPoliciesSchedulingWednesday';

/**
* The NetworksnetworkIdgroupPoliciesScheduling model module.
* @module model/NetworksnetworkIdgroupPoliciesScheduling
* @version 1.0.0
*/
export default class NetworksnetworkIdgroupPoliciesScheduling {
    /**
    * Constructs a new <code>NetworksnetworkIdgroupPoliciesScheduling</code>.
    *     The schedule for the group policy. Schedules are applied to days of the week. 
    * @alias module:model/NetworksnetworkIdgroupPoliciesScheduling
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdgroupPoliciesScheduling</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdgroupPoliciesScheduling} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdgroupPoliciesScheduling} The populated <code>NetworksnetworkIdgroupPoliciesScheduling</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdgroupPoliciesScheduling();
                        
            
            if (data.hasOwnProperty('enabled')) {
                obj['enabled'] = ApiClient.convertToType(data['enabled'], 'Boolean');
            }
            if (data.hasOwnProperty('monday')) {
                obj['monday'] = NetworksnetworkIdgroupPoliciesSchedulingMonday.constructFromObject(data['monday']);
            }
            if (data.hasOwnProperty('tuesday')) {
                obj['tuesday'] = NetworksnetworkIdgroupPoliciesSchedulingTuesday.constructFromObject(data['tuesday']);
            }
            if (data.hasOwnProperty('wednesday')) {
                obj['wednesday'] = NetworksnetworkIdgroupPoliciesSchedulingWednesday.constructFromObject(data['wednesday']);
            }
            if (data.hasOwnProperty('thursday')) {
                obj['thursday'] = NetworksnetworkIdgroupPoliciesSchedulingThursday.constructFromObject(data['thursday']);
            }
            if (data.hasOwnProperty('friday')) {
                obj['friday'] = NetworksnetworkIdgroupPoliciesSchedulingFriday.constructFromObject(data['friday']);
            }
            if (data.hasOwnProperty('saturday')) {
                obj['saturday'] = NetworksnetworkIdgroupPoliciesSchedulingSaturday.constructFromObject(data['saturday']);
            }
            if (data.hasOwnProperty('sunday')) {
                obj['sunday'] = NetworksnetworkIdgroupPoliciesSchedulingSunday.constructFromObject(data['sunday']);
            }
        }
        return obj;
    }

    /**
    * Whether scheduling is enabled (true) or disabled (false). Defaults to false. If true, the schedule objects for each day of the week (monday - sunday) are parsed.
    * @member {Boolean} enabled
    */
    'enabled' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingMonday} monday
    */
    'monday' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingTuesday} tuesday
    */
    'tuesday' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingWednesday} wednesday
    */
    'wednesday' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingThursday} thursday
    */
    'thursday' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingFriday} friday
    */
    'friday' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingSaturday} saturday
    */
    'saturday' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdgroupPoliciesSchedulingSunday} sunday
    */
    'sunday' = undefined;




}
