// meraki_dashboard_api/src/model/NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges model module.
* @module model/NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges
* @version 1.0.0
*/
export default class NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges {
    /**
    * Constructs a new <code>NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges</code>.
    * @alias module:model/NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges
    * @class
    * @param start {String} The first IP in the reserved range
    * @param end {String} The last IP in the reserved range
    * @param comment {String} A text comment for the reserved range
    */

    constructor(start, end, comment) {
        
        
        this['start'] = start;
        this['end'] = end;
        this['comment'] = comment;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges} The populated <code>NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdappliancestaticRoutesstaticRouteIdReservedIpRanges();
                        
            
            if (data.hasOwnProperty('start')) {
                obj['start'] = ApiClient.convertToType(data['start'], 'String');
            }
            if (data.hasOwnProperty('end')) {
                obj['end'] = ApiClient.convertToType(data['end'], 'String');
            }
            if (data.hasOwnProperty('comment')) {
                obj['comment'] = ApiClient.convertToType(data['comment'], 'String');
            }
        }
        return obj;
    }

    /**
    * The first IP in the reserved range
    * @member {String} start
    */
    'start' = undefined;
    /**
    * The last IP in the reserved range
    * @member {String} end
    */
    'end' = undefined;
    /**
    * A text comment for the reserved range
    * @member {String} comment
    */
    'comment' = undefined;




}
