// meraki_dashboard_api/src/model/NetworksnetworkIdgroupPoliciesBonjourForwarding.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdgroupPoliciesBonjourForwardingRules from './NetworksnetworkIdgroupPoliciesBonjourForwardingRules';

/**
* The NetworksnetworkIdgroupPoliciesBonjourForwarding model module.
* @module model/NetworksnetworkIdgroupPoliciesBonjourForwarding
* @version 1.0.0
*/
export default class NetworksnetworkIdgroupPoliciesBonjourForwarding {
    /**
    * Constructs a new <code>NetworksnetworkIdgroupPoliciesBonjourForwarding</code>.
    * The Bonjour settings for your group policy. Only valid if your network has a wireless configuration.
    * @alias module:model/NetworksnetworkIdgroupPoliciesBonjourForwarding
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdgroupPoliciesBonjourForwarding</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdgroupPoliciesBonjourForwarding} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdgroupPoliciesBonjourForwarding} The populated <code>NetworksnetworkIdgroupPoliciesBonjourForwarding</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdgroupPoliciesBonjourForwarding();
                        
            
            if (data.hasOwnProperty('settings')) {
                obj['settings'] = ApiClient.convertToType(data['settings'], 'String');
            }
            if (data.hasOwnProperty('rules')) {
                obj['rules'] = ApiClient.convertToType(data['rules'], [NetworksnetworkIdgroupPoliciesBonjourForwardingRules]);
            }
        }
        return obj;
    }

    /**
    * How Bonjour rules are applied. Can be 'network default', 'ignore' or 'custom'.
    * @member {module:model/NetworksnetworkIdgroupPoliciesBonjourForwarding.SettingsEnum} settings
    */
    'settings' = undefined;
    /**
    * A list of the Bonjour forwarding rules for your group policy. If 'settings' is set to 'custom', at least one rule must be specified.
    * @member {Array.<module:model/NetworksnetworkIdgroupPoliciesBonjourForwardingRules>} rules
    */
    'rules' = undefined;



    /**
    * Allowed values for the <code>settings</code> property.
    * @enum {String}
    * @readonly
    */
    static SettingsEnum = {
        /**
         * value: "network default"
         * @const
         */
        "network default": "network default",
        /**
         * value: "ignore"
         * @const
         */
        "ignore": "ignore",
        /**
         * value: "custom"
         * @const
         */
        "custom": "custom"    };

}
