// meraki_dashboard_api/src/model/NetworksnetworkIdappliancetrafficShapingrulesRules.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdappliancetrafficShapingrulesDefinitions from './NetworksnetworkIdappliancetrafficShapingrulesDefinitions';
import NetworksnetworkIdappliancetrafficShapingrulesPerClientBandwidthLimits from './NetworksnetworkIdappliancetrafficShapingrulesPerClientBandwidthLimits';

/**
* The NetworksnetworkIdappliancetrafficShapingrulesRules model module.
* @module model/NetworksnetworkIdappliancetrafficShapingrulesRules
* @version 1.0.0
*/
export default class NetworksnetworkIdappliancetrafficShapingrulesRules {
    /**
    * Constructs a new <code>NetworksnetworkIdappliancetrafficShapingrulesRules</code>.
    * @alias module:model/NetworksnetworkIdappliancetrafficShapingrulesRules
    * @class
    * @param definitions {Array.<module:model/NetworksnetworkIdappliancetrafficShapingrulesDefinitions>}     A list of objects describing the definitions of your traffic shaping rule. At least one definition is required. 
    */

    constructor(definitions) {
        
        
        this['definitions'] = definitions;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdappliancetrafficShapingrulesRules</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdappliancetrafficShapingrulesRules} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdappliancetrafficShapingrulesRules} The populated <code>NetworksnetworkIdappliancetrafficShapingrulesRules</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdappliancetrafficShapingrulesRules();
                        
            
            if (data.hasOwnProperty('definitions')) {
                obj['definitions'] = ApiClient.convertToType(data['definitions'], [NetworksnetworkIdappliancetrafficShapingrulesDefinitions]);
            }
            if (data.hasOwnProperty('perClientBandwidthLimits')) {
                obj['perClientBandwidthLimits'] = NetworksnetworkIdappliancetrafficShapingrulesPerClientBandwidthLimits.constructFromObject(data['perClientBandwidthLimits']);
            }
            if (data.hasOwnProperty('dscpTagValue')) {
                obj['dscpTagValue'] = ApiClient.convertToType(data['dscpTagValue'], 'Number');
            }
            if (data.hasOwnProperty('priority')) {
                obj['priority'] = ApiClient.convertToType(data['priority'], 'String');
            }
        }
        return obj;
    }

    /**
    *     A list of objects describing the definitions of your traffic shaping rule. At least one definition is required. 
    * @member {Array.<module:model/NetworksnetworkIdappliancetrafficShapingrulesDefinitions>} definitions
    */
    'definitions' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdappliancetrafficShapingrulesPerClientBandwidthLimits} perClientBandwidthLimits
    */
    'perClientBandwidthLimits' = undefined;
    /**
    *     The DSCP tag applied by your rule. null means 'Do not change DSCP tag'.     For a list of possible tag values, use the trafficShaping/dscpTaggingOptions endpoint. 
    * @member {Number} dscpTagValue
    */
    'dscpTagValue' = undefined;
    /**
    *     A string, indicating the priority level for packets bound to your rule.     Can be 'low', 'normal' or 'high'. 
    * @member {String} priority
    */
    'priority' = undefined;




}
