// meraki_dashboard_api/src/model/NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular model module.
* @module model/NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular
* @version 1.0.0
*/
export default class NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular {
    /**
    * Constructs a new <code>NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular</code>.
    * The bandwidth settings for the &#x27;cellular&#x27; uplink
    * @alias module:model/NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular} The populated <code>NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdappliancetrafficShapinguplinkBandwidthBandwidthLimitsCellular();
                        
            
            if (data.hasOwnProperty('limitUp')) {
                obj['limitUp'] = ApiClient.convertToType(data['limitUp'], 'Number');
            }
            if (data.hasOwnProperty('limitDown')) {
                obj['limitDown'] = ApiClient.convertToType(data['limitDown'], 'Number');
            }
        }
        return obj;
    }

    /**
    * The maximum upload limit (integer, in Kbps). null indicates no limit
    * @member {Number} limitUp
    */
    'limitUp' = undefined;
    /**
    * The maximum download limit (integer, in Kbps). null indicates no limit
    * @member {Number} limitDown
    */
    'limitDown' = undefined;




}
