// meraki_dashboard_api/src/model/NetworksnetworkIdswitchroutingospfAreas.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdswitchroutingospfAreas model module.
* @module model/NetworksnetworkIdswitchroutingospfAreas
* @version 1.0.0
*/
export default class NetworksnetworkIdswitchroutingospfAreas {
    /**
    * Constructs a new <code>NetworksnetworkIdswitchroutingospfAreas</code>.
    * @alias module:model/NetworksnetworkIdswitchroutingospfAreas
    * @class
    * @param areaId {Number} OSPF area ID
    * @param areaName {String} Name of the OSPF area
    * @param areaType {module:model/NetworksnetworkIdswitchroutingospfAreas.AreaTypeEnum} Area types in OSPF. Must be one of: [\"normal\", \"stub\", \"nssa\"]
    */

    constructor(areaId, areaName, areaType) {
        
        
        this['areaId'] = areaId;
        this['areaName'] = areaName;
        this['areaType'] = areaType;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdswitchroutingospfAreas</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdswitchroutingospfAreas} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdswitchroutingospfAreas} The populated <code>NetworksnetworkIdswitchroutingospfAreas</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdswitchroutingospfAreas();
                        
            
            if (data.hasOwnProperty('areaId')) {
                obj['areaId'] = ApiClient.convertToType(data['areaId'], 'Number');
            }
            if (data.hasOwnProperty('areaName')) {
                obj['areaName'] = ApiClient.convertToType(data['areaName'], 'String');
            }
            if (data.hasOwnProperty('areaType')) {
                obj['areaType'] = ApiClient.convertToType(data['areaType'], 'String');
            }
        }
        return obj;
    }

    /**
    * OSPF area ID
    * @member {Number} areaId
    */
    'areaId' = undefined;
    /**
    * Name of the OSPF area
    * @member {String} areaName
    */
    'areaName' = undefined;
    /**
    * Area types in OSPF. Must be one of: [\"normal\", \"stub\", \"nssa\"]
    * @member {module:model/NetworksnetworkIdswitchroutingospfAreas.AreaTypeEnum} areaType
    */
    'areaType' = undefined;



    /**
    * Allowed values for the <code>areaType</code> property.
    * @enum {String}
    * @readonly
    */
    static AreaTypeEnum = {
        /**
         * value: "normal"
         * @const
         */
        "normal": "normal",
        /**
         * value: "stub"
         * @const
         */
        "stub": "stub",
        /**
         * value: "nssa"
         * @const
         */
        "nssa": "nssa"    };

}
