// meraki_dashboard_api/src/model/NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers model module.
* @module model/NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers
* @version 1.0.0
*/
export default class NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers {
    /**
    * Constructs a new <code>NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers</code>.
    * @alias module:model/NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers
    * @class
    * @param host {String} IP address to which the APs will send RADIUS accounting messages
    * @param secret {String} Shared key used to authenticate messages between the APs and RADIUS server
    */

    constructor(host, secret) {
        
        
        this['host'] = host;
        this['secret'] = secret;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers} The populated <code>NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdwirelessssidsnumberRadiusAccountingServers();
                        
            
            if (data.hasOwnProperty('host')) {
                obj['host'] = ApiClient.convertToType(data['host'], 'String');
            }
            if (data.hasOwnProperty('port')) {
                obj['port'] = ApiClient.convertToType(data['port'], 'Number');
            }
            if (data.hasOwnProperty('secret')) {
                obj['secret'] = ApiClient.convertToType(data['secret'], 'String');
            }
        }
        return obj;
    }

    /**
    * IP address to which the APs will send RADIUS accounting messages
    * @member {String} host
    */
    'host' = undefined;
    /**
    * Port on the RADIUS server that is listening for accounting messages
    * @member {Number} port
    */
    'port' = undefined;
    /**
    * Shared key used to authenticate messages between the APs and RADIUS server
    * @member {String} secret
    */
    'secret' = undefined;




}
