// meraki_dashboard_api/src/model/NetworksnetworkIdclientsprovisionPoliciesBySsid0.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdclientsprovisionPoliciesBySsid0 model module.
* @module model/NetworksnetworkIdclientsprovisionPoliciesBySsid0
* @version 1.0.0
*/
export default class NetworksnetworkIdclientsprovisionPoliciesBySsid0 {
    /**
    * Constructs a new <code>NetworksnetworkIdclientsprovisionPoliciesBySsid0</code>.
    * The number for the SSID
    * @alias module:model/NetworksnetworkIdclientsprovisionPoliciesBySsid0
    * @class
    * @param devicePolicy {module:model/NetworksnetworkIdclientsprovisionPoliciesBySsid0.DevicePolicyEnum} The policy to apply to the specified client. Can be 'Allowed', 'Blocked', 'Normal' or 'Group policy'. Required.
    */

    constructor(devicePolicy) {
        
        
        this['devicePolicy'] = devicePolicy;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdclientsprovisionPoliciesBySsid0</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdclientsprovisionPoliciesBySsid0} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdclientsprovisionPoliciesBySsid0} The populated <code>NetworksnetworkIdclientsprovisionPoliciesBySsid0</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdclientsprovisionPoliciesBySsid0();
                        
            
            if (data.hasOwnProperty('devicePolicy')) {
                obj['devicePolicy'] = ApiClient.convertToType(data['devicePolicy'], 'String');
            }
            if (data.hasOwnProperty('groupPolicyId')) {
                obj['groupPolicyId'] = ApiClient.convertToType(data['groupPolicyId'], 'String');
            }
        }
        return obj;
    }

    /**
    * The policy to apply to the specified client. Can be 'Allowed', 'Blocked', 'Normal' or 'Group policy'. Required.
    * @member {module:model/NetworksnetworkIdclientsprovisionPoliciesBySsid0.DevicePolicyEnum} devicePolicy
    */
    'devicePolicy' = undefined;
    /**
    * The ID of the desired group policy to apply to the client. Required if 'devicePolicy' is set to \"Group policy\". Otherwise this is ignored.
    * @member {String} groupPolicyId
    */
    'groupPolicyId' = undefined;



    /**
    * Allowed values for the <code>devicePolicy</code> property.
    * @enum {String}
    * @readonly
    */
    static DevicePolicyEnum = {
        /**
         * value: "Allowed"
         * @const
         */
        "Allowed": "Allowed",
        /**
         * value: "Blocked"
         * @const
         */
        "Blocked": "Blocked",
        /**
         * value: "Normal"
         * @const
         */
        "Normal": "Normal",
        /**
         * value: "Group policy"
         * @const
         */
        "Group policy": "Group policy"    };

}
