// meraki_dashboard_api/src/model/NetworksnetworkIdswitchroutingmulticastDefaultSettings.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdswitchroutingmulticastDefaultSettings model module.
* @module model/NetworksnetworkIdswitchroutingmulticastDefaultSettings
* @version 1.0.0
*/
export default class NetworksnetworkIdswitchroutingmulticastDefaultSettings {
    /**
    * Constructs a new <code>NetworksnetworkIdswitchroutingmulticastDefaultSettings</code>.
    * Default multicast setting for entire network. IGMP snooping and Flood unknown multicast traffic settings are enabled by default.
    * @alias module:model/NetworksnetworkIdswitchroutingmulticastDefaultSettings
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdswitchroutingmulticastDefaultSettings</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdswitchroutingmulticastDefaultSettings} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdswitchroutingmulticastDefaultSettings} The populated <code>NetworksnetworkIdswitchroutingmulticastDefaultSettings</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdswitchroutingmulticastDefaultSettings();
                        
            
            if (data.hasOwnProperty('igmpSnoopingEnabled')) {
                obj['igmpSnoopingEnabled'] = ApiClient.convertToType(data['igmpSnoopingEnabled'], 'Boolean');
            }
            if (data.hasOwnProperty('floodUnknownMulticastTrafficEnabled')) {
                obj['floodUnknownMulticastTrafficEnabled'] = ApiClient.convertToType(data['floodUnknownMulticastTrafficEnabled'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * IGMP snooping setting for entire network
    * @member {Boolean} igmpSnoopingEnabled
    */
    'igmpSnoopingEnabled' = undefined;
    /**
    * Flood unknown multicast traffic setting for entire network
    * @member {Boolean} floodUnknownMulticastTrafficEnabled
    */
    'floodUnknownMulticastTrafficEnabled' = undefined;




}
