// meraki_dashboard_api/src/model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12MV22MV72 from './NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12MV22MV72';
import NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12WE from './NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12WE';
import NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV21MV71 from './NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV21MV71';
import NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X from './NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X';
import NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV32 from './NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV32';

/**
* The NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings model module.
* @module model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings
* @version 1.0.0
*/
export default class NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings {
    /**
    * Constructs a new <code>NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings</code>.
    * Video quality and resolution settings for all the camera models.
    * @alias module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings} The populated <code>NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdcameraqualityRetentionProfilesVideoSettings();
                        
            
            if (data.hasOwnProperty('MV21/MV71')) {
                obj['MV21/MV71'] = NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV21MV71.constructFromObject(data['MV21/MV71']);
            }
            if (data.hasOwnProperty('MV12/MV22/MV72')) {
                obj['MV12/MV22/MV72'] = NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12MV22MV72.constructFromObject(data['MV12/MV22/MV72']);
            }
            if (data.hasOwnProperty('MV32')) {
                obj['MV32'] = NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV32.constructFromObject(data['MV32']);
            }
            if (data.hasOwnProperty('MV12WE')) {
                obj['MV12WE'] = NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12WE.constructFromObject(data['MV12WE']);
            }
            if (data.hasOwnProperty('MV22X/MV72X')) {
                obj['MV22X/MV72X'] = NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X.constructFromObject(data['MV22X/MV72X']);
            }
        }
        return obj;
    }

    /**
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV21MV71} MV21/MV71
    */
    'MV21/MV71' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12MV22MV72} MV12/MV22/MV72
    */
    'MV12/MV22/MV72' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV32} MV32
    */
    'MV32' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV12WE} MV12WE
    */
    'MV12WE' = undefined;
    /**
    * @member {module:model/NetworksnetworkIdcameraqualityRetentionProfilesVideoSettingsMV22XMV72X} MV22X/MV72X
    */
    'MV22X/MV72X' = undefined;




}
