// meraki_dashboard_api/src/model/NetworksnetworkIdgroupPoliciesVlanTagging.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdgroupPoliciesVlanTagging model module.
* @module model/NetworksnetworkIdgroupPoliciesVlanTagging
* @version 1.0.0
*/
export default class NetworksnetworkIdgroupPoliciesVlanTagging {
    /**
    * Constructs a new <code>NetworksnetworkIdgroupPoliciesVlanTagging</code>.
    * The VLAN tagging settings for your group policy. Only available if your network has a wireless configuration.
    * @alias module:model/NetworksnetworkIdgroupPoliciesVlanTagging
    * @class
    */

    constructor() {
        
        
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdgroupPoliciesVlanTagging</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdgroupPoliciesVlanTagging} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdgroupPoliciesVlanTagging} The populated <code>NetworksnetworkIdgroupPoliciesVlanTagging</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdgroupPoliciesVlanTagging();
                        
            
            if (data.hasOwnProperty('settings')) {
                obj['settings'] = ApiClient.convertToType(data['settings'], 'String');
            }
            if (data.hasOwnProperty('vlanId')) {
                obj['vlanId'] = ApiClient.convertToType(data['vlanId'], 'String');
            }
        }
        return obj;
    }

    /**
    * How VLAN tagging is applied. Can be 'network default', 'ignore' or 'custom'.
    * @member {module:model/NetworksnetworkIdgroupPoliciesVlanTagging.SettingsEnum} settings
    */
    'settings' = undefined;
    /**
    * The ID of the vlan you want to tag. This only applies if 'settings' is set to 'custom'.
    * @member {String} vlanId
    */
    'vlanId' = undefined;



    /**
    * Allowed values for the <code>settings</code> property.
    * @enum {String}
    * @readonly
    */
    static SettingsEnum = {
        /**
         * value: "network default"
         * @const
         */
        "network default": "network default",
        /**
         * value: "ignore"
         * @const
         */
        "ignore": "ignore",
        /**
         * value: "custom"
         * @const
         */
        "custom": "custom"    };

}
