// meraki_dashboard_api/src/model/NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets model module.
* @module model/NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets
* @version 1.0.0
*/
export default class NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets {
    /**
    * Constructs a new <code>NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets</code>.
    * @alias module:model/NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets
    * @class
    * @param localSubnet {String} The CIDR notation subnet used within the VPN
    */

    constructor(localSubnet) {
        
        
        this['localSubnet'] = localSubnet;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets} The populated <code>NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdappliancevpnsiteToSiteVpnSubnets();
                        
            
            if (data.hasOwnProperty('localSubnet')) {
                obj['localSubnet'] = ApiClient.convertToType(data['localSubnet'], 'String');
            }
            if (data.hasOwnProperty('useVpn')) {
                obj['useVpn'] = ApiClient.convertToType(data['useVpn'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * The CIDR notation subnet used within the VPN
    * @member {String} localSubnet
    */
    'localSubnet' = undefined;
    /**
    * Indicates the presence of the subnet in the VPN
    * @member {Boolean} useVpn
    */
    'useVpn' = undefined;




}
