// meraki_dashboard_api/src/model/NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations model module.
* @module model/NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations
* @version 1.0.0
*/
export default class NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations {
    /**
    * Constructs a new <code>NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations</code>.
    * @alias module:model/NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations
    * @class
    * @param ssidNumber {Number} SSID for which the user is being authorized
    */

    constructor(ssidNumber) {
        
        
        this['ssidNumber'] = ssidNumber;
        
    }

    /**
    * Constructs a <code>NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations} obj Optional instance to populate.
    * @return {module:model/NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations} The populated <code>NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new NetworksnetworkIdmerakiAuthUsersmerakiAuthUserIdAuthorizations();
                        
            
            if (data.hasOwnProperty('ssidNumber')) {
                obj['ssidNumber'] = ApiClient.convertToType(data['ssidNumber'], 'Number');
            }
            if (data.hasOwnProperty('expiresAt')) {
                obj['expiresAt'] = ApiClient.convertToType(data['expiresAt'], 'String');
            }
        }
        return obj;
    }

    /**
    * SSID for which the user is being authorized
    * @member {Number} ssidNumber
    */
    'ssidNumber' = undefined;
    /**
    * Date for authorization to expire. Default is for authorization to not expire.
    * @member {String} expiresAt
    * @default 'Never'
    */
    'expiresAt' = 'Never';




}
