// meraki_dashboard_api/src/model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules model module.
* @module model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules
* @version 1.0.0
*/
export default class OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules {
    /**
    * Constructs a new <code>OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules</code>.
    * @alias module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules
    * @class
    * @param policy {module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules.PolicyEnum} 'allow' or 'deny' traffic specified by this rule
    * @param protocol {module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules.ProtocolEnum} The type of protocol (must be 'tcp', 'udp', 'icmp' or 'any')
    * @param srcCidr {String} Comma-separated list of source IP address(es) (in IP or CIDR notation), or 'any' (FQDN not supported)
    * @param destCidr {String} Comma-separated list of destination IP address(es) (in IP or CIDR notation) or 'any' (FQDN not supported)
    */

    constructor(policy, protocol, srcCidr, destCidr) {
        
        
        this['policy'] = policy;
        this['protocol'] = protocol;
        this['srcCidr'] = srcCidr;
        this['destCidr'] = destCidr;
        
    }

    /**
    * Constructs a <code>OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules} obj Optional instance to populate.
    * @return {module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules} The populated <code>OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules();
                        
            
            if (data.hasOwnProperty('comment')) {
                obj['comment'] = ApiClient.convertToType(data['comment'], 'String');
            }
            if (data.hasOwnProperty('policy')) {
                obj['policy'] = ApiClient.convertToType(data['policy'], 'String');
            }
            if (data.hasOwnProperty('protocol')) {
                obj['protocol'] = ApiClient.convertToType(data['protocol'], 'String');
            }
            if (data.hasOwnProperty('srcPort')) {
                obj['srcPort'] = ApiClient.convertToType(data['srcPort'], 'String');
            }
            if (data.hasOwnProperty('srcCidr')) {
                obj['srcCidr'] = ApiClient.convertToType(data['srcCidr'], 'String');
            }
            if (data.hasOwnProperty('destPort')) {
                obj['destPort'] = ApiClient.convertToType(data['destPort'], 'String');
            }
            if (data.hasOwnProperty('destCidr')) {
                obj['destCidr'] = ApiClient.convertToType(data['destCidr'], 'String');
            }
            if (data.hasOwnProperty('syslogEnabled')) {
                obj['syslogEnabled'] = ApiClient.convertToType(data['syslogEnabled'], 'Boolean');
            }
        }
        return obj;
    }

    /**
    * Description of the rule (optional)
    * @member {String} comment
    */
    'comment' = undefined;
    /**
    * 'allow' or 'deny' traffic specified by this rule
    * @member {module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules.PolicyEnum} policy
    */
    'policy' = undefined;
    /**
    * The type of protocol (must be 'tcp', 'udp', 'icmp' or 'any')
    * @member {module:model/OrganizationsorganizationIdappliancevpnvpnFirewallRulesRules.ProtocolEnum} protocol
    */
    'protocol' = undefined;
    /**
    * Comma-separated list of source port(s) (integer in the range 1-65535), or 'any'
    * @member {String} srcPort
    */
    'srcPort' = undefined;
    /**
    * Comma-separated list of source IP address(es) (in IP or CIDR notation), or 'any' (FQDN not supported)
    * @member {String} srcCidr
    */
    'srcCidr' = undefined;
    /**
    * Comma-separated list of destination port(s) (integer in the range 1-65535), or 'any'
    * @member {String} destPort
    */
    'destPort' = undefined;
    /**
    * Comma-separated list of destination IP address(es) (in IP or CIDR notation) or 'any' (FQDN not supported)
    * @member {String} destCidr
    */
    'destCidr' = undefined;
    /**
    * Log this rule to syslog (true or false, boolean value) - only applicable if a syslog has been configured (optional)
    * @member {Boolean} syslogEnabled
    */
    'syslogEnabled' = undefined;



    /**
    * Allowed values for the <code>policy</code> property.
    * @enum {String}
    * @readonly
    */
    static PolicyEnum = {
        /**
         * value: "allow"
         * @const
         */
        "allow": "allow",
        /**
         * value: "deny"
         * @const
         */
        "deny": "deny"    };
    /**
    * Allowed values for the <code>protocol</code> property.
    * @enum {String}
    * @readonly
    */
    static ProtocolEnum = {
        /**
         * value: "tcp"
         * @const
         */
        "tcp": "tcp",
        /**
         * value: "udp"
         * @const
         */
        "udp": "udp",
        /**
         * value: "icmp"
         * @const
         */
        "icmp": "icmp",
        /**
         * value: "any"
         * @const
         */
        "any": "any"    };

}
