// meraki_dashboard_api/src/model/OrganizationsorganizationIdadminsTags.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The OrganizationsorganizationIdadminsTags model module.
* @module model/OrganizationsorganizationIdadminsTags
* @version 1.0.0
*/
export default class OrganizationsorganizationIdadminsTags {
    /**
    * Constructs a new <code>OrganizationsorganizationIdadminsTags</code>.
    * @alias module:model/OrganizationsorganizationIdadminsTags
    * @class
    * @param tag {String} The name of the tag
    * @param access {module:model/OrganizationsorganizationIdadminsTags.AccessEnum} The privilege of the dashboard administrator on the tag. Can be one of 'full', 'read-only', 'guest-ambassador' or 'monitor-only'
    */

    constructor(tag, access) {
        
        
        this['tag'] = tag;
        this['access'] = access;
        
    }

    /**
    * Constructs a <code>OrganizationsorganizationIdadminsTags</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/OrganizationsorganizationIdadminsTags} obj Optional instance to populate.
    * @return {module:model/OrganizationsorganizationIdadminsTags} The populated <code>OrganizationsorganizationIdadminsTags</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OrganizationsorganizationIdadminsTags();
                        
            
            if (data.hasOwnProperty('tag')) {
                obj['tag'] = ApiClient.convertToType(data['tag'], 'String');
            }
            if (data.hasOwnProperty('access')) {
                obj['access'] = ApiClient.convertToType(data['access'], 'String');
            }
        }
        return obj;
    }

    /**
    * The name of the tag
    * @member {String} tag
    */
    'tag' = undefined;
    /**
    * The privilege of the dashboard administrator on the tag. Can be one of 'full', 'read-only', 'guest-ambassador' or 'monitor-only'
    * @member {module:model/OrganizationsorganizationIdadminsTags.AccessEnum} access
    */
    'access' = undefined;



    /**
    * Allowed values for the <code>access</code> property.
    * @enum {String}
    * @readonly
    */
    static AccessEnum = {
        /**
         * value: "full"
         * @const
         */
        "full": "full",
        /**
         * value: "read-only"
         * @const
         */
        "read-only": "read-only",
        /**
         * value: "guest-ambassador"
         * @const
         */
        "guest-ambassador": "guest-ambassador",
        /**
         * value: "monitor-only"
         * @const
         */
        "monitor-only": "monitor-only"    };

}
