// meraki_dashboard_api/src/model/OrganizationsorganizationIdappliancesecurityintrusionAllowedRules.js
/**
 * Meraki Dashboard API
 * The Cisco Meraki Dashboard API is a modern REST API based on the OpenAPI specification.  > Date: 05 August, 2020 > > [What's New](https://meraki.io/whats-new/)  ---  [API Documentation](https://meraki.io/api)  [Community Support](https://meraki.io/community)  [Meraki Homepage](https://www.meraki.com) 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
* The OrganizationsorganizationIdappliancesecurityintrusionAllowedRules model module.
* @module model/OrganizationsorganizationIdappliancesecurityintrusionAllowedRules
* @version 1.0.0
*/
export default class OrganizationsorganizationIdappliancesecurityintrusionAllowedRules {
    /**
    * Constructs a new <code>OrganizationsorganizationIdappliancesecurityintrusionAllowedRules</code>.
    * @alias module:model/OrganizationsorganizationIdappliancesecurityintrusionAllowedRules
    * @class
    * @param ruleId {String} A rule identifier of the format meraki:intrusion/snort/GID/<gid>/SID/<sid>. gid and sid can be obtained from either https://www.snort.org/rule-docs or as ruleIds from the security events in /organization/[orgId]/securityEvents
    */

    constructor(ruleId) {
        
        
        this['ruleId'] = ruleId;
        
    }

    /**
    * Constructs a <code>OrganizationsorganizationIdappliancesecurityintrusionAllowedRules</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/OrganizationsorganizationIdappliancesecurityintrusionAllowedRules} obj Optional instance to populate.
    * @return {module:model/OrganizationsorganizationIdappliancesecurityintrusionAllowedRules} The populated <code>OrganizationsorganizationIdappliancesecurityintrusionAllowedRules</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OrganizationsorganizationIdappliancesecurityintrusionAllowedRules();
                        
            
            if (data.hasOwnProperty('ruleId')) {
                obj['ruleId'] = ApiClient.convertToType(data['ruleId'], 'String');
            }
            if (data.hasOwnProperty('message')) {
                obj['message'] = ApiClient.convertToType(data['message'], 'String');
            }
        }
        return obj;
    }

    /**
    * A rule identifier of the format meraki:intrusion/snort/GID/<gid>/SID/<sid>. gid and sid can be obtained from either https://www.snort.org/rule-docs or as ruleIds from the security events in /organization/[orgId]/securityEvents
    * @member {String} ruleId
    */
    'ruleId' = undefined;
    /**
    * Message is optional and is ignored on a PUT call. It is allowed in order for PUT to be compatible with GET
    * @member {String} message
    */
    'message' = undefined;




}
